---
title: "Reduced levels of relatedness indicate that great-tailed grackles disperse further at the edge of their range"
format: pdf
editor: source
author: 
- 'Lukas, Dieter^1^*'
- 'Blackwell, Aaron D^2^'
- 'Edrisi, Maryam^2^'
- 'Hardy, Kristin^3,4^'
- 'Marfori, Zara^1^'
- 'McCune, Kelsey^5,6^'
- 'Sevchik, August^7,8^'
- 'Smith, Caroline^2^'
- 'Logan, Corina J^1,5^'
bibliography: MyLibrary.bib
csl: https://raw.githubusercontent.com/citation-style-language/styles/master/apa.csl
urlcolor: blue
header-includes:
  - \usepackage[left]{lineno}
  - \linenumbers
---

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE) 
#Make code wrap text so it doesn't go off the page when Knitting to PDF

knitr::opts_chunk$set(echo=F, include=T, results='asis', warning=F, message=F) 
#sets global options to display code along with the results https://exeter-data-analytics.github.io/LitProg/r-markdown.html
#set echo=F for knitting to PDF (hide code), and echo=T for knitting to HTML (show code)
```

Open... ![](logoOpenAccess.png){width="5%"} access ![](logoOpenCode.png){width="5%"} [code](https://github.com/corinalogan/grackles/blob/master/Files/Preregistrations/gxpopbehaviorhabitatq2dispersal.qmd) ![](logoOpenPeerReview.png){width="5%"} peer review

**Affiliations:** 1) Max Planck Institute for Evolutionary Anthropology; Leipzig, Germany, 2) Washington State University, Pullman; USA, 3) University of Rochester; Rochester, USA, 4) Current affiliation: University of California Davis; Davis, USA, 5) University of California Santa Barbara; Santa Barbara, USA, 6) Current affiliation: Auburn University; Auburn, USA, 7) Arizona State University; Tempe, USA, 8) Current affiliation: Florida Atlantic University; Boca Raton, USA, \*Corresponding author: dieter_lukas\@eva.mpg.de

**This article is based on the following preregistration:** Logan CJ, McCune KB, Chen N, Lukas D. 2020. [**Implementing a rapid geographic range expansion - the role of behavior and habitat changes**](http://corinalogan.com/Preregistrations/gxpopbehaviorhabitat.html) 

**which has been pre-study peer reviewed and received an In Principle Recommendation by:** Esther Sebastián González (2020) The role of behavior and habitat availability on species geographic expansion. *Peer Community in Ecology*, 100062. [10.24072/pci.ecology.100062](https://doi.org/10.24072/pci.ecology.100062). Reviewers: Caroline Nieberding, Tim Parker, and Pizza Ka Yee Chow

![](logoPCIecology.png){width="50%"}

### ABSTRACT

It is generally thought that behavioral flexibility, the ability to change behavior when circumstances change, plays an important role in the ability of a species to rapidly expand their geographic range. However, it is an alternative non-exclusive possibility that an increase in the amount of available habitat can also facilitate a range expansion. Great-tailed grackles (*Quiscalus mexicanus*) are a social, polygamous species that is rapidly expanding its geographic range and eats a variety of human foods in addition to foraging on insects and on the ground for other natural food items. They are behaviorally flexible and highly associated with human-modified environments, thus offering an opportunity to assess the role of behavior and habitat change over the course of their expansion. Here, we compare behavior in wild-caught grackles from two populations across their range (a more recent population in the middle of the northern expansion front in Arizona versus a very recent population on the northern edge of the expansion front in California) to investigate whether individuals in a more recently established population exhibit more dispersal behavior (i.e., individuals are more likely to move away from their parents). We find that levels of relatedness are lower in the population closer to the edge compared to the population nearer the core. In particular, we observe no closely related individuals at the edge, suggesting that individuals of both sexes disperse further in this population than in the population nearer the core. Our analyses also suggest that, in both populations, females generally move shorter distances from where they hatched than males. These results elucidate that the rapid geographic range expansion of great-tailed grackles is associated with individuals differentially expressing dispersal behaviors.

### INTRODUCTION

It is generally thought that behavioral flexibility, the ability to change behavior when circumstances change [see @mikhalevich_is_2017 for theoretical background on our flexibility definition], plays an important role in the ability of a species to rapidly expand their geographic range [e.g., @lefebvre1997feeding; @griffin2014innovation; @chow2016practice; @sol2000behavioural; @sol2002behavioural; @sol2005big; @sol2007big]. These ideas predict that flexibility, exploration, and innovation facilitate the expansion of individuals into completely new areas and that their role diminishes after a certain number of generations [@wright2010behavioral]. In support of this, experimental studies have shown that latent abilities are primarily expressed in a time of need [e.g., @taylor2007spontaneous; @bird2009insightful; @manrique2011spontaneous; @auersperg2012spontaneous; @laumer2018spontaneous]. Therefore, we do not expect the founding individuals who initially dispersed out of their original range to have unique behavioral characteristics that are passed on to their offspring. Instead, we expect that the actual act of continuing a range expansion relies on flexibility, exploration, innovation, persistence, and dispersal, and that these behaviors are therefore expressed more on the edge of the expansion range where there have not been many generations to accumulate relevant knowledge about the environment.

To determine whether behavior is involved in a rapid geographic range expansion, direct measures of individual behavioral abilities must be collected in populations across the range of the species [see the discussion on the danger of proxies of flexibility in @logan2018beyond]. We tested whether dispersal  might have played a role in the rapid geographic range expansion of great-tailed grackles (*Quiscalus mexicanus*). Great-tailed grackles are behaviorally flexible [@logan2016flexibilityproblem], rapidly expanding their geographic range [@wehtje2003range], and highly associated with human-modified environments [@johnson2001great], thus offering an opportunity to assess the role of behavior and habitat change over the course of their expansion. This social, polygamous species eats a variety of human foods in addition to foraging on insects and on the ground for other natural food items [@johnson2001great]. 

We here compare behavior in wild-caught great-tailed grackles from two populations across their range (a more recent population in the middle of the northern expansion front in Arizona versus a very recent population on the northern edge of the expansion front in California). We investigate whether certain behaviors are expressed differently in the edge population relative to older populations. There could be multiple mechanisms underpinning the results we find, however our aim is to narrow down the role of changes in behavior and changes in habitats in the range expansion of great-tailed grackles. Results will elucidate whether the rapid geographic range expansion of great-tailed grackles is associated with individuals differentially expressing particular behaviors and/or whether the expansion is facilitated by the alignment of their natural behaviors with an increase in suitable habitat (i.e., human-modified environments). The preregistration associated with our article set out multiple hypotheses for how behavior could be linked to the rapid range expansion of great-tailed grackles. We previously completed the research and published the results linked to the hypotheses on habitat changes [@summers2023xpop], behavioral flexibility, innovativeness, exploration, and persistence across the range of great-tailed grackles [@logan2023xpoppcj] and species differences between great-tailed and boat-tailed grackles [@logan2024xpopbtgr].

In this article, we compare the dispersal behavior of great-tailed grackles between a recently established population (California) and a population that has existed for several generations (Arizona)(Table 1). We examine whether individuals in a recently established population (California) are more likely to move away from the location where they hatched by determining whether their average relatedness (calculated using single nucleotide polymorphisms, SNPs) is lower than what we would expect if individuals move randomly [@sevchik2021dispersal]. Genetic approaches are one of the main ways to determine patterns of dispersal in birds, because actual dispersal events are rare and difficult to study. In most species, individuals only show limited movement from their place of origin to where they themselves breed, creating patterns of isolation by distance as pairs of individuals are less likely to share genetic variants the further away they are [@manel2003landscape]. We measure this sharing of genetic variants using relatedness [@spong2001deriving], with individuals who show low relatedness to others in the population being assumed to have moved further [@aguillon2017deconstructing]. Though we refer to our two sampling sites as two populations, it is important to note that the distribution of great-tailed grackles appears continuous and connected. Even at the edge, the expansion appears to occur gradually, rather than through the establishment of separate, distanced populations [@wehtje2003range]. Accordingly, our focus is not on the overall levels of genetic diversity or relatedness in the full sample, but how genetic variation at each site is structured according to the sex of individuals and the distances between them to lead to patterns of relatedness.


### RESEARCH QUESTION

Our research question is listed as it appeared in the preregistration.

#### Q2 (dispersal behavior: great-tailed grackles): Are there differences in dispersal behavior across the great-tailed grackle's geographic range? (Fig. 1, Table 1)

**Prediction 2:** We predict **more dispersal at the edge**: a higher proportion of individuals, particularly females, which is the sex that appears to disperse less in the population in the middle of the range expansion [@sevchik2021dispersal], disperse in a more recently established population and, accordingly, fewer individuals are closely related to each other. *This would support the hypothesis* that changes in dispersal behavior are involved in the great-tailed grackle's geographic range expansion.

**Prediction 2 alternative 1:** We predict that the **proportion of individuals dispersing is not related to when the population established** at a particular site and, accordingly, the average relatedness is similar across populations. *This supports the hypothesis* that the original dispersal behavior was already well adapted in this species to facilitate a range expansion.

**Table 1.** Population characteristics for each of the two field sites. The number of generations at a site is based on a generation length of 5.6 years for this species [@GTGRbirdlife2018, note that this species starts breeding at age 1], and on the first year in which this species was reported (or estimated) to breed at each location. Woodland, California: Yolo Audubon Society's newsletter *The Burrowing Owl* (July 2004), which Steve Hampton shared with Logan; Tempe, Arizona: estimated based on 1945 first-sighting report in nearby Phoenix, Arizona [@wehtje2004thesis] to which we added 6 years, which is the average time between first-sighting and first-breeding - see Table 3 in [@wehtje2003range].


```{r eval=TRUE, warning=FALSE, results='asis', include=TRUE}
#| label: table1
#| tbl-colwidths: [60,60,30,30,30,60]
d <- read.csv(url("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/gxpopbehaviorhabitat_Table1.csv"), header=F, sep=",", stringsAsFactors=F) 

d<-d[2:3,]
colnames(d) <- c("Site","Range position","Breeding since","Number of years breeding","Average number of generations","Citation")
rownames<-c(1,2)

knitr::kable(d, row.names = FALSE)
```

### STATE OF THE DATA AND CHANGES FROM PREREGISTRATION

The preregistration was written (March 2020) prior to collecting any data from the edge population, therefore we were blind to these data. However, we were not blind to some of the data from the Arizona population: some of the relatedness data (SNPs used for Hypothesis 2 to quantify relatedness to infer whether individuals disperse away from relatives) from the middle population (Arizona) had already been analyzed for other purposes [n=57 individuals, see @sevchik2021dispersal]. Therefore, we consider it secondary data: data that were collected for other investigations. We collected blood samples from many more grackles in Arizona, and we repeated the analyses for the Arizona population with the complete sample. This preregistration was submitted in May 2020 to PCI Ecology for pre-study peer review. We received the reviews, and revised and resubmitted in August 2020, and it passed pre-study peer review in October 2020.

While our ideal plan was to include three field sites, due to restrictions around COVID-19 and because we learned about potential risks to the safety of study participants at the initially considered third field site, it was not possible for us to accomplish all of our goals within our current funding period. We therefore compare only two populations.


### METHODS

#### Sample

**Q2:** Great-tailed grackles were caught in the wild in Tempe, Arizona, and in Woodland and Sacramento, California. Adults were identified from their eye color, which changes from brown to yellow upon reaching adulthood [@johnson2001great]. We applied colored leg bands in unique combinations for individual identification. Some individuals (~20) were brought temporarily into aviaries for behavioral choice tests, and then were released back to the wild at their point of capture. We caught grackles with a variety of methods (e.g., walk-in traps, mist nets, bow nets), some of which decrease the likelihood of a selection bias for exploratory and bold individuals because grackles cannot see the traps (i.e., mist nets).


#### Sample size rationale

We tested as many great-tailed grackles as we could during the two to three years we spent at each site given that the birds are only brought into the aviaries during the non-breeding season (approximately September through March). It is time intensive to conduct the aviary test battery (2-6 months per bird at the Arizona field site), therefore we approximated that the minimum sample size at each site will follow the minimum sample sizes in Table 2 with the aim that half of the grackles tested at each site are female. We sampled more than the expected 20 grackles per site for the genetic analyses.


#### Protocols and open materials

DNA was collected from the grackles, processed, and analyzed for pairwise relatedness using ddRADseq and Stacks as in @sevchik2021dispersal ([protocol](http://corinalogan.com/Preregistrations/gdispersal_manuscript.html#Blood_collection)). Our pre-registration only included a brief summary of the methods, we describe them in detail below for full reproducibility.

We previously generated genotypes for 57 individuals from Arizona in 2018 [@sevchik2021dispersal]. For the current analyses, we added genotypes for 37 individuals from Arizona and 35 individuals from California. In brief, we collected 150uL of blood from individual birds by brachial or medial metatarsal venipuncture. Samples were centrifuged at 15x gravity for 10 minutes directly after collection to separate the serum from the cellular fraction. The serum layer was removed and 600uL of lysis buffer was added to the remaining packed cells. Tubes containing packed cells and lysis buffer were stored at room temperature for up to 5 years before DNA extraction. The time gap between sampling and extraction could have reduced the quality of the samples, as we observed that some samples had clotted. This potentially explains the reduced SNP recovery rate for some of the samples (see below). DNA was extracted from the samples using the DNeasy Blood and Tissue kit (Qiagen). Extracted DNA samples were shipped with ice packs to the Cornell Lab of Ornithology for ddRAD sequencing in August 2023. The sequencing to generate single-nucleotide polymorphism (SNP; where at a given position in the genome two different bases, alleles, can occur) genotypes was performed at the Cornell University Lab of Ornithology. Fragments were digested with a combination of two restriction enzymes (SbfI-HF and MspI), cleaned, size-selected, amplified using a low-cycle PCR process, and pooled together for sequencing on an Illumina NextSeq500. 

We performed the SNP processing and selection as in @thrasher2018double, processing the samples from the two populations separately. For Arizona, we combined the genotypes of the individuals sequenced in 2018 with the genotypes sequenced in 2023 prior to the processing so that all individuals have the same set of alleles to compare for the relatedness estimation. Occurrences of rare alleles are likely to differ among the two separate populations, therefore combining the data from the two populations could potentially lead to ascertainment biases, where alleles that occur in the population with the larger sample but not in the population with the smaller sample are included as informative whereas alleles that occur only in the population with the smaller sample are excluded. This would bias the relatedness estimation in the smaller population because differences among individuals in this population would be lost. The re-processing means that, for those individuals already included in @sevchik2021dispersal, the genotypes, and the resulting pairwise relatedness estimates, are slightly different compared to those previously estimated. For both populations, loci were considered only if they were present in 95% of the samples (r) and had a minimum minor allele frequency of 0.05 (min maf).

To prepare, check, and describe the genotype data (expected heterozygosity, probability of identity), we used functions in the R packages ‘adegenet’ [@Jombart2008adegenet], ‘pegas’ [@Paradis2010pegas], and ‘popgenutils [@Tourvas2020popgenutils].

For each population, we calculated the pairwise relatedness among all dyads of adult individuals using the estimator by Queller & Goodnight (1989), which was more robust for our inferences in a subset of the Arizona data [@sevchik2021dispersal], as implemented in the package ‘related’ [@pew2015related] in R. For the relatedness calculation, we only used the genotypes of individuals in the respective population to derive the allele frequencies that set the baseline chance of sharing alleles. That means that, overall, in both populations average relatedness will be close to zero. Individuals who share fewer alleles than expected have a negative relatedness value, while individuals who share more alleles than expected have a positive value. Our focus is not on comparing the overall levels of genetic diversity in the two populations, but whether there is structure in the sharing of alleles that lead to patterns of relatedness among individuals of the same sex. We identified as potential kin those pairs of individuals whose estimated relatedness was equal to or larger than 0.25 (closer relatives, at the level of half-siblings) or larger than 0.125 (distant relatives, at the level of cousins). 

We recorded the longitude and latitude of the first locations where individuals were observed after they had been caught and released, or for those individuals who were not resighted, the location where they were trapped. To calculate the geographic distance among pairs of individuals based on these locations, we used the function ‘distm’ in the package 'geosphere' [@hijmans2022geosphere] with the Vincenty ellipsoid great circle function. 


#### Open data

All data for analyses are available at Edmond [@lukas2024dispersal]. Raw genotype files are available in the Sequence Read Archive of the National Center for Biotechnology Information (NCBI, accession number: PRJNA658480).

#### Blinding during analysis

Blinding is usually not involved in the final analyses because the experimenters collect the data (and therefore have seen some form of it) and run the analyses. However, when processing the genetic data, the experimenters and the people who conducted the lab work were blind to the relatedness amongst the birds.

### ANALYSIS

We did not exclude any data except for instances where missing data made analyses not reliable. Samples with a low DNA quantity and quality produce data for only a small number of SNP loci. Relatedness estimates are only reliable if they are based on several hundred SNP loci [wang2016pedigrees, foroughirad2019quality], because small numbers of loci can lead to high variances in the estimates. Analyses were conducted in R [current version `r getRversion()`; @rcoreteam] and Stan [version 2.18, @carpenter2017stan]. We used functions in the package ‘rethinking’ [@mcelreath2020statistical] to construct and summarize the linear models. Following the social convention of this approach, we report the 89% compatibility intervals (89% CI) of the posterior sample.

Our response variable is the average relatedness between all pairs of individuals within one sex. As in @sevchik2021dispersal, we analysed this in two ways: first, as a continuous variable ranging between -1 and +1, reflecting average relatedness as whether individuals share more or less alleles than expected by chance; and second, as a categorical variable coded as yes/no, reflecting whether the average relatedness among a pair of individuals is more or less than the threshold that kin are expected to have (r≧0.125 and r≧0.25). We had planned to include as explanatory variables the site diameter, the site sample size, and the number of generations at a site. However, because we were able to only obtain samples from two populations, we did not include these variables in the models because it would be impossible to say which of the factors might explain the site differences (see also our Discussion). We did however use the site diameter data to ensure that the two populations were comparable. Permutations (i.e., randomly assigning site ID to individuals) and general linear models estimating average relatedness of each individual to all others at that site (averagerelatedness ~ $\alpha$[site]) were used to determine whether individuals at one site are more closely related to each other than the individuals at another site. 

#### Comparison of average relatedness between the two populations

We compared the overall levels of average relatedness, as well as the average relatedness among the females and among the males, between the population in Arizona and the population in California using a linear model:


$$
\begin{aligned}
pairwiserelatedness_{d} \sim Normal(\mu_{d},\theta)\\
\mu_{d} = \alpha_{pop[d]} + \beta_{sex[d],pop[d]}\\
\alpha_{pop[d]} \sim Normal(0,1)\\
\begin{bmatrix}
\beta_{d,1}\\
\beta_{d,2}
\end{bmatrix}
\sim \textrm{MVNormal}\begin{pmatrix} \begin{bmatrix}
0\\
0
\end{bmatrix}
, S_{sex}
\end{pmatrix}\\
S =
\begin{pmatrix}
  \sigma_{sex=1} & 0\\
  0 & \sigma_{sex=2}
\end{pmatrix}
R
\begin{pmatrix}
  \sigma_{sex=1} & 0\\
  0 & \sigma_{sex=2}
\end{pmatrix}\\
R \sim LKJcorr(4)\\
\sigma_{sex} \sim Exponential(1)\\
\theta \sim Exponential(1)\\
\end{aligned}
$$

where the $pairwiserelatedness_{d}$, the relatedness among all pairs of relatives in the two populations $pop$ of either $sex$, is assumed to be distributed according to a normal distribution with mean $\mu$ and variance $\theta$. We assumed that relatedness overall might be higher in one population than the other, and therefore included an interaction between population and sex, such that the intercepts are defined by a two dimensional Gaussian distribution ($MVNormal$) with means of 0, because we separately include the population means as $\alpha_{pop}$, and covariance matrices $S$ reflecting the two $sexes$. The covariance matrix, $S$, is factored into separate standard deviations, $\sigma_{sex}$, and a correlation matrix, $R$. The prior for the correlation matrix is set to come from the Lewandowski-Kurowicka-Joe ($LKJcorr$) distribution, and is set to be weakly informative and skeptical of extreme correlations near -1 or 1.


#### Comparison of degree of kinship between the two populations
We compared the number of of individuals classified as either close or distant relatives in the two populations using a binomial model:

$$
\begin{aligned}
kin_{d} \sim Binomial(1,p_{d})\\
logit(p_{d}) = \alpha_{pop[d]} + \beta_{sex[d],pop[d]}\\
\alpha_{pop[d]} \sim Normal(0,1)\\
\begin{bmatrix}
\beta_{d,1}\\
\beta_{d,2}
\end{bmatrix}
\sim \textrm{MVNormal}\begin{pmatrix} \begin{bmatrix}
0\\
0
\end{bmatrix}
, S_{sex}
\end{pmatrix}\\
\end{aligned}
$$
$$
\begin{aligned}
S =
\begin{pmatrix}
  \sigma_{sex=1} & 0\\
  0 & \sigma_{sex=2}
\end{pmatrix}
R
\begin{pmatrix}
  \sigma_{sex=1} & 0\\
  0 & \sigma_{sex=2}
\end{pmatrix}\\
R \sim LKJcorr(4)\\
\sigma_{sex} \sim Exponential(1)\\
\end{aligned}
$$

where the $kin_{d}$ reflects whether the relatedness of a given pair of individuals is or is not larger than the threshold for either close or distant relatives. All remaining terms as above.

Second, we compared the presence of kin in the two populations using permutations. Average relatedness declines the more individuals are included in the calculation [@lukas2005extent]. Permutations are a way to account for this by assessing whether any observed differences remain when comparing the same number of individuals. We randomly took 10,000 draws of the same number of individuals we had in the California population, which was a smaller sample, from the genotypes we had in the Arizona population (e.g., randomly drawing 13 of the female genotypes in Arizona and calculating the number of kin observed in this sample, before repeating the random draw another 9999 times, each time calculating the number of kin observed in the sample). We then compared the observed number of kin in California to the numbers obtained in the 10,000 random samples to assess whether the kinship composition in California is similar or different to that observed in Arizona 


#### Sex biases in dispersal in the two populations
To determine whether, in either or both populations, individuals of one sex are more likely to disperse farther than individuals of the opposite sex, we first compared the average relatedness among females to the average relatedness among males in the same population. We performed 10,000 random draws, drawing the same number of individuals from the whole population as there are females or males in that population, to assess whether the relatedness among individuals of one sex is different than that observed in a random sample of individuals of the same size from that population. Next, we determined the geographic distances among those pairs of individuals identified as potential close or distant kin. We again performed 10,000 draws, drawing the same number as there are kin of that sex from all the females or males in that population to assess the expected distance among such a sample of same-sex individuals. If the distances among the 10,000 draws are generally larger than those observed among kin, then we infer that kin of that sex remain closer together than what would be expected by chance. Finally, we performed assessments of spatial autocorrelation to link the pairwise relatedness among individuals of each sex to the geographic distances of their locations. 

To test whether males and females show different patterns of genetic isolation by geographic distance, we followed analyses as in @aguillon2017deconstructing. In each population, for males and females separately, we assessed the strength of the association between the the matrices of average relatedness and of geographic distance using Mantel correlograms with the function ‘mantel.correlog’ in the ‘vegan’ package [@oksanen2013package] in R. For each of the four associations (two sexes in two populations), we performed 10,000 permutations to assess the strength of the association. The approach involves partitioning the geographic locations into a series of discrete distance classes. We used two methods to create the distance classes. First, we attempted to have about equal numbers of pairs of individuals within each distance class, creating nine distance classes of (0-100m, 100-200m, 200-300m, 300-400m, 400-500m, 500-750m, 750-1000m,1 000-1250m, and 1250-2000m). With the second method, we only created two distance classes to increase the sample size in each class, splitting the distance according to the limit at which most close kin were detected (0-400m and 400-1400m). For each distance class, a normalized Mantel statistic is calculated using permutations of values within that distance class. The permutation statistics, plotted against distance classes, produce a multivariate correlogram. A negative correlation between genetic relatedness and spatial distance indicates that the more closely related individuals are found closer to each other, indicating that these individuals likely disperse shorter distances than those individuals where a positive correlation is found.


### RESULTS

#### Summary statistics

*California SNP data*\
We retained 493 SNPs. Data was missing for 3.3% of all alleles (individuals missing information for either one or both of their chromosomes for that particular position). None of the SNPs showed a particular underrepresentation of information. The missingness was due to the incomplete genotype of one individual (C116RY, adult male), who had missing data at 459 of the 493 SNPs (93%), whereas all other individuals had data missing at four or fewer SNPs. We excluded this individual from the further analyses, because relatedness calculations based on so few SNPs were, as expected, highly stochastic and led to extreme deviations (see code chunk ‘kin composition’ in the Rmd file for illustration). For the remaining individuals, all SNPs had two alleles and the observed heterozygosity (individuals carrying one copy each of the two bases) was 0.29, identical to the heterozygosity expected in a population with the same allele frequencies and random mating. The probability of identity for siblings, the chance that two siblings will show the same genotypes given the allele frequencies across these 493 SNP loci and random mating among individuals, is less than $10^{-64}$. This indicates that any relatedness we detect among individuals is likely to reflect biological relatedness, rather than resulting from limited sampling making individuals more similar.

*Arizona SNP data*\
We retained 462 SNPs. Data was missing for 3.0% of all alleles (individuals missing information for either one or both of their chromosomes for that particular position). None of the SNPs showed a particular underrepresentation of information. There were three individuals whose genotypes were less complete (A072KB, adult female, missing data at 191 (41%) of SNPs; A088YR, adult male, missing data at 174 (38%) of SNPs; A059NB, adult female, at 148 (32%) of SNPs), whereas all other individuals had data missing at less than 10% of SNPs. Here, we did not exclude any individuals because the number of SNPs with information was still sufficiently high for all genotypes to reduce the noise in the relatedness estimation. All SNPs had two alleles and the observed heterozygosity (individuals carrying one copy each of the two bases) was 0.29, similar to the heterozygosity expected in a population with the same allele frequencies and random mating. The probability of identity for siblings, the chance that two siblings will show the same genotypes given the allele frequencies across these 462 SNP loci and random mating among individuals, is less than $10^{-60}$.  


```{r descriptive statistics, eval=FALSE, include=FALSE, echo=FALSE}
library(adegenet)
library(pegas)
library(dplyr)

# Overall sample size
samplesize<-as.data.frame(adultids %>% group_by(Population,Sex) %>% summarize(samplesize=n()))

# missing data in the genotypes
# California
sum(ca_snpdata==0)
sum(ca_snpdata==0)/(nrow(ca_snpdata)*(ncol(ca_snpdata)-1)) # 3.3% missing data
apply(ca_snpdata[,2:986], 1, function(x) length(which(x=="0")))

# Arizona
sum(az_snpdata==0)
sum(az_snpdata==0)/(nrow(az_snpdata)*(ncol(az_snpdata)-1)) # 3.0% missing data
apply(az_snpdata[,2:925], 1, function(x) length(which(x=="0")))

# remove Buñuelo C116RY because of missing data


ca_genepopfile<-read.table("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/g_flexforaging_data_genepop_Woodland.txt",sep=" ",colClasses="character")

az_genepopfile<-read.table("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/g_flexforaging_data_genepop_Tempe.txt",sep=" ",colClasses="character")


ca_genepopfile_ade<-ca_genepopfile
row.names(ca_genepopfile_ade)<-ca_genepopfile_ade[,1]
ca_genepopfile_ade<-ca_genepopfile_ade[,2:ncol(ca_genepopfile_ade)]
counter<-1
ca_genepopfile_ade_joined<-as.data.frame(matrix(ncol=ncol(ca_genepopfile_ade)/2,nrow=nrow(ca_genepopfile_ade)))
for(i in 1:(ncol(ca_genepopfile_ade)/2)){
  ca_genepopfile_ade_joined[,i]<-paste(ca_genepopfile_ade[,counter], ca_genepopfile_ade[,counter+1], sep="")
  counter<-counter+2
}
ca_ade_input<-df2genind(ca_genepopfile_ade_joined, ncode=2,NA.char="00")
ca_ade_hw<-hw.test(ca_ade_input)
ca_toto <- summary(ca_ade_input)
bartlett.test(list(ca_toto$Hexp,ca_toto$Hobs))
median(ca_toto$Hexp-ca_toto$Hobs)
mean(ca_toto$Hexp)
mean(ca_toto$Hobs)
sum(ca_ade_hw[,4]<0.005)/nrow(ca_ade_hw)
ca_toto$NA.perc

apply(ca_genepopfile_ade_joined[,1:493], 1, function(x) length(which(x=="0000")))
apply(ca_genepopfile_ade_joined, 2, function(x) length(which(x=="0000")))

az_genepopfile_ade<-az_genepopfile
row.names(az_genepopfile_ade)<-az_genepopfile_ade[,1]
az_genepopfile_ade<-az_genepopfile_ade[,2:ncol(az_genepopfile_ade)]
counter<-1
az_genepopfile_ade_joined<-as.data.frame(matrix(ncol=ncol(az_genepopfile_ade)/2,nrow=nrow(az_genepopfile_ade)))
for(i in 1:(ncol(az_genepopfile_ade)/2)){
  az_genepopfile_ade_joined[,i]<-paste(az_genepopfile_ade[,counter], az_genepopfile_ade[,counter+1], sep="")
  counter<-counter+2
}
az_ade_input<-df2genind(az_genepopfile_ade_joined, ncode=2,NA.char="00")
az_ade_hw<-hw.test(az_ade_input)
az_toto <- summary(az_ade_input)
bartlett.test(list(az_toto$Hexp,az_toto$Hobs))
median(az_toto$Hexp-az_toto$Hobs)
mean(az_toto$Hexp)
mean(az_toto$Hobs)
sum(az_ade_hw[,4]<0.005)/nrow(az_ade_hw)
ca_toto$NA.perc

apply(az_genepopfile_ade_joined[,1:462], 1, function(x) length(which(x=="0000")))
apply(az_genepopfile_ade_joined, 2, function(x) length(which(x=="0000")))


P_id_sibs<-NA
for(i in 2:987){
  if(i %%2 == 0){
  p<-(sum(ca_snpdata[,i]==1)+sum(ca_snpdata[,i+1]==1))/(2*nrow(ca_snpdata)-(sum(ca_snpdata[,i]==0)+sum(ca_snpdata[,i+1]==0)))
  q<-(sum(ca_snpdata[,i]==2)+sum(ca_snpdata[,i+1]==2))/(2*nrow(ca_snpdata)-(sum(ca_snpdata[,i]==0)+sum(ca_snpdata[,i+1]==0)))
  r<-(sum(ca_snpdata[,i]==3)+sum(ca_snpdata[,i+1]==3))/(2*nrow(ca_snpdata)-(sum(ca_snpdata[,i]==0)+sum(ca_snpdata[,i+1]==0)))
  s<-(sum(ca_snpdata[,i]==4)+sum(ca_snpdata[,i+1]==4))/(2*nrow(ca_snpdata)-(sum(ca_snpdata[,i]==0)+sum(ca_snpdata[,i+1]==0)))
  P_id_sibs[(i/2)] <- 0.25 + (0.5*sum(p*p+q*q+r*r+s*s))+(0.5*(p*p+q*q+r*r+s*s)^2)-(0.25*(p^4+q^4+r^4+s^4))
  i<-i+1
  }
}

cumprod(P_id_sibs)
ca_P_id_sibs<-cumprod(P_id_sibs)[493]

P_id_sibs<-NA
for(i in 2:925){
  if(i %%2 == 0){
  p<-(sum(az_snpdata[,i]==1)+sum(az_snpdata[,i+1]==1))/(2*nrow(az_snpdata)-(sum(az_snpdata[,i]==0)+sum(az_snpdata[,i+1]==0)))
  q<-(sum(az_snpdata[,i]==2)+sum(az_snpdata[,i+1]==2))/(2*nrow(az_snpdata)-(sum(az_snpdata[,i]==0)+sum(az_snpdata[,i+1]==0)))
  r<-(sum(az_snpdata[,i]==3)+sum(az_snpdata[,i+1]==3))/(2*nrow(az_snpdata)-(sum(az_snpdata[,i]==0)+sum(az_snpdata[,i+1]==0)))
  s<-(sum(az_snpdata[,i]==4)+sum(az_snpdata[,i+1]==4))/(2*nrow(az_snpdata)-(sum(az_snpdata[,i]==0)+sum(az_snpdata[,i+1]==0)))
  P_id_sibs[(i/2)] <- 0.25 + (0.5*sum(p*p+q*q+r*r+s*s))+(0.5*(p*p+q*q+r*r+s*s)^2)-(0.25*(p^4+q^4+r^4+s^4))
  i<-i+1
  }
}

cumprod(P_id_sibs)
az_P_id_sibs<-cumprod(P_id_sibs)[462]

```

*Sample*\
In total, we included genotype information for 52 females and 27 males in Arizona, and 13 females and 15 males in California. In Arizona, all birds were found within a maximum of 1,991m from each other (median 669m). In California, birds were found at multiple locations. Twelve females and twelve males were found at a location in Sacramento that spanned about the same range as the population in Arizona (maximum geographic distance 1,592m, median 474m). Three birds (one female, two males) were resighted at a separate location in Sacramento ~7,000m away from the main location. In addition, one male was trapped and resighted at a location ~33,000m away in Woodland. Therefore, the maximum and average geographic distances between the locations of individuals are much higher for the California sample than the Arizona sample. For the set of analyses that include pairwise geographic distances among individuals, we performed the analyses only with the birds found at the single location in Sacramento in order to keep the California population comparable to the Arizona population (i.e., we excluded these four birds). 

```{r sample load data and calculate pairwise relatedness, eval=FALSE, include=FALSE, echo=FALSE}
#Load the relevant packages
library(related)
library(tidyr)
library(dplyr)
library(vegan)
library(geosphere)
library(rethinking)


#Load the relevant data
#SNP data, processed to be in correct format to calculate pairwise relatedness (output files saved for flexforaging article)
ca_input<-readgenotypedata("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/g_flexforaging_data_genepop_Woodland.txt")

az_input<-readgenotypedata("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/g_flexforaging_data_genepop_Tempe.txt")

# Load additional information on individuals
idinfo <- read.csv(url("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/gxpopbehaviorhabitatq2dispersal_individualinfoforrelatedness.csv"), header=T, sep=",", stringsAsFactors=F)

# remove an individual for which we did not have blood, and therefore no genotype
idinfo<-idinfo[-38,]

# remove Buñuelo C116RY because of missing data
idinfo<-idinfo[idinfo$BirdID != "C116RY",]

# Exclude juveniles from the dataset, because they might not yet have dispersed
adultids<-idinfo[idinfo$Age %in% "A",]

#identify which individuals are female and which are male
females<-filter(adultids,Sex %in% "F",Age %in% "A")[,]$BirdID
males<-filter(adultids,Sex %in% "M",Age %in% "A")[,]$BirdID

samplesize<-as.data.frame(adultids %>% group_by(Population,Sex) %>% summarize(samplesize=n()))

ca_snpdata<-ca_input$gdata
ca_snpdata$V1<-gsub("-","",ca_snpdata$V1)
ca_snpdata<-ca_snpdata[ca_snpdata$V1 %in% adultids$BirdID,]

az_snpdata<-az_input$gdata
az_snpdata$V1<-gsub("A-","A",az_snpdata$V1)
az_snpdata<-az_snpdata[az_snpdata$V1 %in% adultids$BirdID,]

# individual A053PS is included twice in the genetic dataset - because we resequenced this individual
az_snpdata<-az_snpdata[-45,] # remove original
# Need to re-arrange the data frame now to make sure that this individual is in the right place
az_snpdata<-arrange(az_snpdata,V1)

# az_snpdata<-az_snpdata[-54,] # remove new, as alternative because the reprocessing appears to have fixed the previous problems

#Calculate pairwise relatedness, here choosing the relatedness method developed by Wang, Queller & Goodknight, Lynch & Li
ca_outfile<-coancestry(ca_snpdata,wang=1,quellergt=1,lynchli=1)
az_outfile<-coancestry(az_snpdata,wang=1,quellergt=1,lynchli=1)

#extract the relevant information from the file
ca_pairwise_r<-ca_outfile$relatedness
az_pairwise_r<-az_outfile$relatedness

# List for each individuals their pairwise relatedness to all other members of the group
ca_completerelatedness<-as.data.frame(matrix(nrow=2*nrow(ca_pairwise_r),ncol=ncol(ca_pairwise_r)))
colnames(ca_completerelatedness)<-colnames(ca_pairwise_r)
ca_completerelatedness$pair.no<-c(ca_pairwise_r$pair.no,ca_pairwise_r$pair.no)
ca_completerelatedness$ind1.id<-c(ca_pairwise_r$ind1.id,ca_pairwise_r$ind2.id)
ca_completerelatedness$ind2.id<-c(ca_pairwise_r$ind2.id,ca_pairwise_r$ind1.id)
ca_completerelatedness$group<-c(ca_pairwise_r$group,ca_pairwise_r$group)
ca_completerelatedness$trioml<-c(ca_pairwise_r$trioml,ca_pairwise_r$trioml)
ca_completerelatedness$wang<-c(ca_pairwise_r$wang,ca_pairwise_r$wang)
ca_completerelatedness$lynchli<-c(ca_pairwise_r$lynchli,ca_pairwise_r$lynchli)
ca_completerelatedness$lynchrd<-c(ca_pairwise_r$lynchrd,ca_pairwise_r$lynchrd)
ca_completerelatedness$ritland<-c(ca_pairwise_r$ritland,ca_pairwise_r$ritland)
ca_completerelatedness$quellergt<-c(ca_pairwise_r$quellergt,ca_pairwise_r$quellergt)
ca_completerelatedness$dyadml<-c(ca_pairwise_r$dyadml,ca_pairwise_r$dyadml)



az_completerelatedness<-as.data.frame(matrix(nrow=2*nrow(az_pairwise_r),ncol=ncol(az_pairwise_r)))
colnames(az_completerelatedness)<-colnames(az_pairwise_r)
az_completerelatedness$pair.no<-c(az_pairwise_r$pair.no,az_pairwise_r$pair.no)
az_completerelatedness$ind1.id<-c(az_pairwise_r$ind1.id,az_pairwise_r$ind2.id)
az_completerelatedness$ind2.id<-c(az_pairwise_r$ind2.id,az_pairwise_r$ind1.id)
az_completerelatedness$group<-c(az_pairwise_r$group,az_pairwise_r$group)
az_completerelatedness$trioml<-c(az_pairwise_r$trioml,az_pairwise_r$trioml)
az_completerelatedness$wang<-c(az_pairwise_r$wang,az_pairwise_r$wang)
az_completerelatedness$lynchli<-c(az_pairwise_r$lynchli,az_pairwise_r$lynchli)
az_completerelatedness$lynchrd<-c(az_pairwise_r$lynchrd,az_pairwise_r$lynchrd)
az_completerelatedness$ritland<-c(az_pairwise_r$ritland,az_pairwise_r$ritland)
az_completerelatedness$quellergt<-c(az_pairwise_r$quellergt,az_pairwise_r$quellergt)
az_completerelatedness$dyadml<-c(az_pairwise_r$dyadml,az_pairwise_r$dyadml)

az_individualrelatedness<-as.data.frame(az_completerelatedness %>% group_by(ind1.id) %>% summarise(average_r=mean(quellergt)))

# Check distribution of relatedness values
hist(az_completerelatedness$wang)
hist(ca_completerelatedness$wang)

```





```{r distance calculation, include=FALSE, echo=FALSE, eval=FALSE}
# Distance between trap sites (meters): straight line distance (assuming earth as an ellipsoid) between all pairs of trapping locations based on the longitude and latitude of each site.

#Plot pairwise distances among all females and among all males in the sample
#Calculate all pairwise distances
all_pairwise_distances <- distm(adultids[,c('Lon','Lat')], adultids[,c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(all_pairwise_distances)<-adultids$Name
colnames(all_pairwise_distances)<-adultids$Name
diag(all_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult females in California
CA_female_pairwise_distances <- distm(adultids[adultids$Sex=="F" & adultids$Population=="Woodland",c('Lon','Lat')], adultids[adultids$Sex=="F" & adultids$Population=="Woodland",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(CA_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Woodland",]$BirdID
colnames(CA_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Woodland",]$BirdID
diag(CA_female_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult females in Arizona
AZ_female_pairwise_distances <- distm(adultids[adultids$Sex=="F" & adultids$Population=="Tempe",c('Lon','Lat')], adultids[adultids$Sex=="F" & adultids$Population=="Tempe",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(AZ_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Tempe",]$BirdID
colnames(AZ_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Tempe",]$BirdID
diag(AZ_female_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult males in California
CA_male_pairwise_distances <- distm(adultids[adultids$Sex=="M" & adultids$Population=="Woodland",c('Lon','Lat')], adultids[adultids$Sex=="M" & adultids$Population=="Woodland",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(CA_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Woodland",]$BirdID
colnames(CA_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Woodland",]$BirdID
diag(CA_male_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult males in Arizona
AZ_male_pairwise_distances <- distm(adultids[adultids$Sex=="M" & adultids$Population=="Tempe",c('Lon','Lat')], adultids[adultids$Sex=="M" & adultids$Population=="Tempe",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(AZ_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Tempe",]$BirdID
colnames(AZ_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Tempe",]$BirdID
diag(AZ_male_pairwise_distances)<-NA

#plot distributions of pairwise distances
hist(all_pairwise_distances,col="grey75",border="black",breaks=10)
hist(CA_female_pairwise_distances,col="grey75",border="black",breaks=10)
hist(CA_male_pairwise_distances,col="grey75",border="black",breaks=20)
hist(AZ_female_pairwise_distances,col="grey75",border="black",breaks=10)
hist(AZ_male_pairwise_distances,col="grey75",border="black",breaks=20)


AZ_all_pairwise_distance<-all_pairwise_distances[rownames(all_pairwise_distances) %in% adultids[adultids$Population == "Tempe",]$Name,colnames(all_pairwise_distances) %in% adultids[adultids$Population == "Tempe",]$Name]

CA_all_pairwise_distance<-all_pairwise_distances[rownames(all_pairwise_distances) %in% adultids[adultids$Population == "Woodland",]$Name,colnames(all_pairwise_distances) %in% adultids[adultids$Population == "Woodland",]$Name]

# In California, one individual was trapped and resighted at a location in Woodland (C124YA Ak'xi'), while the remaining birds were found at locations in Sacramento. Therefore, the maximum distance between the locations of individuals is much higher in California than it is in Arizona, where all birds were captured in locations around Tempe. In addition, three birds (C127OS Talingo, C128SS Bacmut bacni, and C135YS Quiscalus) were resighted at a separate location in Woodland. For the analyses including pairwise distances, we will run the analyses with all California birds, and with only the birds found at a single location in Sacramento. 
hist(CA_all_pairwise_distance)
# Distances for the birds in California that are found at a single location
CA_all_pairwise_distance_singlelocation<-CA_all_pairwise_distance[c(-1,-3,-19,-21),c(-1,-3,-19,-21)]
hist(CA_all_pairwise_distance_singlelocation)
# For comparison, pairwise distances among all birds in Arizona
hist(AZ_all_pairwise_distance)

CA_female_pairwise_distances_singlelocation<-CA_female_pairwise_distances[-10,-10]
CA_male_pairwise_distances_singlelocation<-CA_male_pairwise_distances[c(-1,-2,-10),c(-1,-2,-10)]

```




#### Difference in dispersal behavior between the two populations

*Comparison of average relatedness in the two populations*\
Overall, the average relatedness among individuals in the two populations is slightly negative, which is more pronounced in California (average relatedness: Arizona -0.013, California -0.037). This slight skew toward negative relatedness values suggests that both populations, but particularly the population in California, might contain individuals who have immigrated into these populations and are therefore sharing fewer alleles than would be expected by chance. In Arizona, males (-0.009) have slightly higher average relatedness than females (-0.013). In California, females (-0.024) have higher average relatedness than males (-0.048). 

The model comparing levels of pairwise relatedness between the two populations indicates that the values in California are consistently lower than the values in Arizona because their confidence intervals do not cross zero (median of contrast for females -0.003, 89%CI of contrast -0.006 to -0.001, for males median -0.009, 89% CI -0.011 to -0.007).



```{r compare average relatedness, include=FALSE, echo=FALSE, eval=FALSE}

#Calculate average of and variance in relatedness among all individuals, all females, and all males
# using the relatedness estimates based on the method by Queller and Goodnight
mean(filter(ca_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
mean(filter(ca_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
mean(ca_completerelatedness$quellergt)

var(filter(ca_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
var(filter(ca_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
var(ca_completerelatedness$quellergt)


mean(filter(az_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
mean(filter(az_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
mean(az_completerelatedness$quellergt)

var(filter(az_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
var(filter(az_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
var(az_completerelatedness$quellergt)
```



```{r linear models comparing relatedness in the two populations, include=FALSE, echo=FALSE, eval=FALSE}

birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
az_completerelatedness<-left_join(az_completerelatedness,birdsex,by="ind1.id")
ca_completerelatedness<-left_join(ca_completerelatedness,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
az_completerelatedness<-left_join(az_completerelatedness,birdsex,by="ind2.id")
ca_completerelatedness<-left_join(ca_completerelatedness,birdsex,by="ind2.id")
az_completerelatedness$sexdyad<-paste(az_completerelatedness$ind1.sex,az_completerelatedness$ind2.sex)
ca_completerelatedness$sexdyad<-paste(ca_completerelatedness$ind1.sex,ca_completerelatedness$ind2.sex)


# Compare whether average relatedness is higher in one of the populations

az_completerelatedness[which(az_completerelatedness$sexdyad=="F F"),]$sexdyad<-"1"
az_completerelatedness[which(az_completerelatedness$sexdyad=="M F"),]$sexdyad<-"2"
az_completerelatedness[which(az_completerelatedness$sexdyad=="F M"),]$sexdyad<-"2"
az_completerelatedness[which(az_completerelatedness$sexdyad=="M M"),]$sexdyad<-"3"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="F F"),]$sexdyad<-"1"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="M F"),]$sexdyad<-"2"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="F M"),]$sexdyad<-"2"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="M M"),]$sexdyad<-"3"

dat_relatedness<-list(
  relatedness = c(az_completerelatedness$quellergt,ca_completerelatedness$quellergt),
  population = as.numeric(as.factor(c(rep(1,nrow(az_completerelatedness)),rep(2,nrow(ca_completerelatedness))))),
  sex = as.numeric(as.factor(c(az_completerelatedness$sexdyad,ca_completerelatedness$sexdyad)))
)

set.seed(1109)
mimmodel <- ulam( alist(
  relatedness ~dnorm(nu,theta),
  nu <- p[population]+s[sex,population],
  p[population]~dnorm(0,1),
  vector[2]:s[sex] ~ multi_normal(0,Rho_s,sigma_s),
  Rho_s~dlkjcorr(4),
  sigma_s~dexp(1),
  theta ~ dexp(1)
) , data=dat_relatedness, chains=4 , cores=4 , cmdstan = TRUE, iter=4000)

precis(mimmodel,depth=3)
# p[1]       -0.01 0.07 -0.03  0.00 1.02   223.40
# p[2]       -0.04 0.04 -0.08  0.00 1.01   989.07
# s[1,1]     -0.01 0.07 -0.02  0.02 1.02   220.88
# s[2,1]     -0.01 0.07 -0.02  0.01 1.02   219.32
# s[3,1]      0.00 0.07 -0.01  0.02 1.02   209.79
# s[1,2]      0.01 0.04 -0.03  0.05 1.01   989.73
# s[2,2]      0.00 0.04 -0.04  0.04 1.01   978.47
# s[3,2]     -0.01 0.04 -0.05  0.03 1.01   984.99
# Rho_s[1,1]  1.00 0.00  1.00  1.00   NA       NA
# Rho_s[2,1] -0.02 0.35 -0.57  0.54 1.01  1537.44
# Rho_s[1,2] -0.02 0.35 -0.57  0.54 1.01  1537.44
# Rho_s[2,2]  1.00 0.00  1.00  1.00   NA       NA
# sigma_s[1]  0.03 0.11  0.00  0.08 1.01   187.24
# sigma_s[2]  0.04 0.08  0.01  0.14 1.00   713.60
# theta       0.08 0.00  0.07  0.08 1.00  3364.24

posterior_mim<-extract.samples(mimmodel)

# Population 1 = Arizona, Population 2 = California; contrast 2-1: negative values mean that Arizona has higher relatedness than California
populationcontrast_mir<-inv_logit(posterior_mim$p[,2])-inv_logit(posterior_mim$p[,1])
precis(populationcontrast_mir)
# populationcontrast_mir -0.01 0.02 -0.02  0.01  ▁▁▁▁▇▁▁▁

females_populationcontrast_mim<-inv_logit(posterior_mim$s[,1,2]+posterior_mim$p[,2])-inv_logit(posterior_mim$s[,1,1]+posterior_mim$p[,1])
HPDI(females_populationcontrast_mim)
#        5.5          94.5
# -0.0059496279 -0.0009124577 

males_populationcontrast_mim<-inv_logit(posterior_mim$s[,3,2]+posterior_mim$p[,2])-inv_logit(posterior_mim$s[,3,1]+posterior_mim$p[,1])
HPDI(males_populationcontrast_mim)
#        5.5          94.5
# -0.011467572 -0.006740249

opp_populationcontrast_mim<-inv_logit(posterior_mim$s[,2,2]+posterior_mim$p[,2])-inv_logit(posterior_mim$s[,2,1]+posterior_mim$p[,1])
HPDI(opp_populationcontrast_mim)
#        0.89        0.89 
# -0.007068200 -0.003930558 
```


*Comparison of the degree of likely kin in the two populations*\
Overall, in both populations we identified very low numbers of dyads that are potentially kin (Figure 1). In California, none of the dyads are estimated to be related with r≧0.25, and only one opposite sex dyad is estimated to be related with r≧0.125 (out of 105 male-male dyads, 78 female-female dyads, and 195 opposite sex dyads). In Arizona, 3 male-male dyads (0.9% of the 351 male-male dyads), 12 opposite sex dyads (0.9% of the 1404 dyads), and 9 female-female dyads (0.7% of 1326 dyads) are estimated to be related with r≧0.25. With the lower threshold of r≧0.125, 9 (2.5%) of all male-male dyads, 39 (2.9%) of female-female dyads, and 32 (2.3%) of opposite sex dyads in Arizona are classified as related. 

A binomial model indicates that the probability that any dyad would be kin at r≧0.25 is higher among individuals in Arizona than in California (median posterior estimate of difference in probabilities for close kin r≧0.25 0.6%, 89% CI 0.4 to 1.0%, for more distant kin r≧0.125 median difference 2.2%, 89% CI 1.6 to 2.9%). The differences in probability hold for both female-female dyads (r≧0.25: 0.7, 89% CI 0.3 to 1.0%; r≧0.125: 2.5%, 89% CI 1.7 to 3.3%) and for male-male dyads (r≧0.25: 0.6%, 89% CI 0.2 to 1.1%; r≧0.125: 2.3%, 89% CI 1.2 to 3.2%). 

The permutations support that the absence of same-sex individuals related at r≧0.125 in the California population is not simply due to the smaller sample of individuals. There are no relatives in only 12% of permutations drawing 13 individuals from among the 52 females in Arizona, and there are no relatives in only 2% of permutations drawing 15 individuals from among the 27 males in Arizona.



```{r kin composition, include=FALSE, echo=FALSE, eval=FALSE}

# Determine number of dyads that are kin in the two populations
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
az_completerelatedness<-left_join(az_completerelatedness,birdsex,by="ind1.id")
ca_completerelatedness<-left_join(ca_completerelatedness,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
az_completerelatedness<-left_join(az_completerelatedness,birdsex,by="ind2.id")
ca_completerelatedness<-left_join(ca_completerelatedness,birdsex,by="ind2.id")
az_completerelatedness$sexdyad<-paste(az_completerelatedness$ind1.sex,az_completerelatedness$ind2.sex)
ca_completerelatedness$sexdyad<-paste(ca_completerelatedness$ind1.sex,ca_completerelatedness$ind2.sex)


ca_completerelatedness$closekin_qg<-ca_completerelatedness$quellergt>0.249
ca_completerelatedness$kin_qg<-ca_completerelatedness$quellergt>0.1249

az_completerelatedness$closekin_qg<-az_completerelatedness$quellergt>0.249
az_completerelatedness$kin_qg<-az_completerelatedness$quellergt>0.1249


# Quick check of the classification with the Wang estimator when including the individual with the missing data
ca_completerelatedness %>% group_by(sexdyad) %>% summarise(sum(closekin_w))
#  F F                     0
#  F M                     3
#  M M                     4
ca_completerelatedness[ca_completerelatedness$closekin_w==T,]
# Highlights again that we should use Queller & Goodknight instead of Wang estimator - the close kin dyads with the Wang estimator involve C116RY, the individual with the missing data.



# California, close kin - values displayed below are already adjusted for the double counting of dyads. There are 378 dyads in total, value in brackets reflects percentage, second value in brackets reflects percentage of dyads of that sex who are kin (78 FF dyads, 195 FM dyads, 105 MM dyads)
ca_completerelatedness %>% group_by(sexdyad) %>% summarise(sum(closekin_qg))
#  F F                      0 (0, 0)
#  F M                      0 (0, 0)
#  M M                      0 (0, 0)
ca_completerelatedness %>% group_by(sexdyad) %>% summarise(sum(kin_qg))
#  F F                 0 (0, 0)
#  F M                 1 (0.002645503, 0.005128205)
#  M M                 0 (0, 0) 

# Arizona, close kin - values displayed below are already adjusted for the double counting of dyads. There are 3081 dyads in total, value in brackets reflects percentage, second value in brackets reflects percentage of dyads of that sex who are kin (1326 FF dyads, 1404 FM dyads, 351 MM dyads)
az_completerelatedness %>% group_by(sexdyad) %>% summarise(sum(closekin_qg))
#  F F                      9 (0.00292113, 0.00678733)
#  F M                     12 (0.003894839, 0.008547009)
#  M M                      3 (0.0009737098, 0.008547009)
az_completerelatedness %>% group_by(sexdyad) %>% summarise(sum(kin_qg))
#  F F                39 (0.01265823, 0.02941176)
#  F M                32 (0.01038624, 0.02279202)
#  M M                9 (0.00292113, 0.02564103)


```






```{r linear model kin composition, include=FALSE, echo=FALSE, eval=FALSE}


# Compare whether number of close kin is higher in one of the populations

# Add information on the sex of the individuals to the relatedness data
az_arecloserelatives<-az_pairwise_r
az_arecloserelatives$closerelatives<-as.integer(az_pairwise_r$quellergt>0.249)
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
az_arecloserelatives <-left_join(az_arecloserelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
az_arecloserelatives <-left_join(az_arecloserelatives,birdsex,by="ind2.id")

ca_arecloserelatives<-ca_pairwise_r
ca_arecloserelatives$closerelatives<-as.integer(ca_pairwise_r$quellergt>0.249)
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
ca_arecloserelatives <-left_join(ca_arecloserelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
ca_arecloserelatives <-left_join(ca_arecloserelatives,birdsex,by="ind2.id")

az_arecloserelatives$sexdyad<-paste(az_arecloserelatives$ind1.sex,az_arecloserelatives$ind2.sex)
ca_arecloserelatives$sexdyad<-paste(ca_arecloserelatives$ind1.sex,ca_arecloserelatives$ind2.sex)
az_arecloserelatives[which(az_arecloserelatives$sexdyad=="F F"),]$sexdyad<-"1"
az_arecloserelatives[which(az_arecloserelatives$sexdyad=="M F"),]$sexdyad<-"2"
az_arecloserelatives[which(az_arecloserelatives$sexdyad=="F M"),]$sexdyad<-"2"
az_arecloserelatives[which(az_arecloserelatives$sexdyad=="M M"),]$sexdyad<-"3"
ca_arecloserelatives[which(ca_arecloserelatives$sexdyad=="F F"),]$sexdyad<-"1"
ca_arecloserelatives[which(ca_arecloserelatives$sexdyad=="M F"),]$sexdyad<-"2"
ca_arecloserelatives[which(ca_arecloserelatives$sexdyad=="F M"),]$sexdyad<-"2"
ca_arecloserelatives[which(ca_arecloserelatives$sexdyad=="M M"),]$sexdyad<-"3"


dat_closerelatives<-list(
  relative = c(az_arecloserelatives$closerelatives,ca_arecloserelatives$closerelatives),
  population = as.numeric(as.factor(c(rep(1,nrow(az_arecloserelatives)),rep(2,nrow(ca_arecloserelatives))))),
  sex = as.numeric(as.factor(c(az_arecloserelatives$sexdyad,ca_arecloserelatives$sexdyad)))
)

# simpler model as originally planned

mipmodel <- ulam( alist(
  relative ~dbinom(1,p),
  logit(p) <- pop[population],
  pop[population]~dnorm(-5,3)
) , data=dat_closerelatives, chains=4 , cores=4 , cmdstan = TRUE, iter=4000)

precis(mipmodel,depth=2)

posterior_mip<-extract.samples(mipmodel)
populationcontrast_mip<-inv_logit(posterior_mip$pop[,2])-inv_logit(posterior_mip$pop[,1])
HPDI(populationcontrast_mip)
# contrasts (pop 1=Arizona, pop 2=California)
#  more likely to have relatives in Arizona 
# -0.010240430 -0.004036319 
sum(populationcontrast_mip<0)
# 7954/8000 of posterior samples >0 (99%)
mean(populationcontrast_mip)
# -0.006897721


# More detailed model that takes into account sex

set.seed(1109)
mirmodel <- ulam( alist(
  relative ~dbinom(1,p),
  logit(p) <- pop[population]+s[sex,population],
  pop[population]~dnorm(-5,3),
  vector[2]:s[sex] ~ multi_normal(0,Rho_s,sigma_s),
  Rho_s~dlkjcorr(4),
  sigma_s~dexp(1)
) , data=dat_closerelatives, chains=4 , cores=4 , cmdstan = TRUE, iter=4000)

precis(mirmodel,depth=3)

# contrasts (pop 1=Arizona, pop 2=California; s 1=FF, s2=FM, s3=MM)
posterior_mir<-extract.samples(mirmodel)
populationcontrast_mir<-inv_logit(posterior_mir$pop[,2])-inv_logit(posterior_mir$pop[,1])
precis(populationcontrast_mir)
HPDI(populationcontrast_mir)
# populationcontrast_mir -0.01 0.01 -0.01     0   ▁▁▁▁▇▁▁
# -0.01088361 -0.00255777 

females_populationcontrast_mir<-inv_logit(posterior_mir$s[,1,2]+posterior_mir$pop[,2])-inv_logit(posterior_mir$s[,1,1]+posterior_mir$pop[,1])
precis(females_populationcontrast_mir)
HPDI(females_populationcontrast_mir)
# no difference in the likelihood that female-female dyads are closely related
#                                 mean   sd  5.5% 94.5%  histogram
# females_populationcontrast_mir    0 0.01 -0.01  0.01 ▁▁▇▁▁▁▁▁▁
# -0.011763689  0.003416313

males_populationcontrast_mir<-inv_logit(posterior_mir$s[,3,2]+posterior_mir$pop[,2])-inv_logit(posterior_mir$s[,3,1]+posterior_mir$pop[,1])
precis(males_populationcontrast_mir)
HPDI(males_populationcontrast_mir)
# no difference in the likelihood that male-male dyads are closely related
#                              mean   sd  5.5% 94.5%       histogram
# males_populationcontrast_mir -0.01 0.01 -0.02     0 ▁▁▁▃▇▁▁▁▁▁▁▁
# -0.017703086  0.002763918

opp_populationcontrast_mir<-inv_logit(posterior_mir$s[,2,2]+posterior_mir$pop[,2])-inv_logit(posterior_mir$s[,2,1]+posterior_mir$pop[,1])
precis(opp_populationcontrast_mir)
HPDI(opp_populationcontrast_mir)
# less likely to have opposite sex relatives in California
#                            mean   sd  5.5% 94.5%  histogram
# opp_populationcontrast_mir -0.01  0 -0.01     0 ▁▂▇▂▁▁▁▁▁
# -0.012964160 -0.002121524




# simple model for distant relatives

az_aredistantrelatives<-az_pairwise_r
az_aredistantrelatives$distantrelatives<-as.integer(az_pairwise_r$quellergt>0.1249)
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
az_aredistantrelatives <-left_join(az_aredistantrelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
az_aredistantrelatives <-left_join(az_aredistantrelatives,birdsex,by="ind2.id")

ca_aredistantrelatives<-ca_pairwise_r
ca_aredistantrelatives$distantrelatives<-as.integer(ca_pairwise_r$quellergt>0.1249)
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
ca_aredistantrelatives <-left_join(ca_aredistantrelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
ca_aredistantrelatives <-left_join(ca_aredistantrelatives,birdsex,by="ind2.id")

az_aredistantrelatives$sexdyad<-paste(az_aredistantrelatives$ind1.sex,az_aredistantrelatives$ind2.sex)
ca_aredistantrelatives$sexdyad<-paste(ca_aredistantrelatives$ind1.sex,ca_aredistantrelatives$ind2.sex)
az_aredistantrelatives[which(az_aredistantrelatives$sexdyad=="F F"),]$sexdyad<-"1"
az_aredistantrelatives[which(az_aredistantrelatives$sexdyad=="M F"),]$sexdyad<-"2"
az_aredistantrelatives[which(az_aredistantrelatives$sexdyad=="F M"),]$sexdyad<-"2"
az_aredistantrelatives[which(az_aredistantrelatives$sexdyad=="M M"),]$sexdyad<-"3"
ca_aredistantrelatives[which(ca_aredistantrelatives$sexdyad=="F F"),]$sexdyad<-"1"
ca_aredistantrelatives[which(ca_aredistantrelatives$sexdyad=="M F"),]$sexdyad<-"2"
ca_aredistantrelatives[which(ca_aredistantrelatives$sexdyad=="F M"),]$sexdyad<-"2"
ca_aredistantrelatives[which(ca_aredistantrelatives$sexdyad=="M M"),]$sexdyad<-"3"


dat_distantrelatives<-list(
  relative = c(az_aredistantrelatives$distantrelatives,ca_aredistantrelatives$distantrelatives),
  population = as.numeric(as.factor(c(rep(1,nrow(az_aredistantrelatives)),rep(2,nrow(ca_aredistantrelatives))))),
  sex = as.numeric(as.factor(c(az_aredistantrelatives$sexdyad,ca_aredistantrelatives$sexdyad)))
)

# simpler model as originally planned

mipmodel <- ulam( alist(
  relative ~dbinom(1,p),
  logit(p) <- pop[population],
  pop[population]~dnorm(-5,3)
) , data=dat_distantrelatives, chains=4 , cores=4 , cmdstan = TRUE, iter=4000)

precis(mipmodel,depth=2

posterior_mip<-extract.samples(mipmodel)
populationcontrast_mip<-inv_logit(posterior_mip$pop[,2])-inv_logit(posterior_mip$pop[,1])
HPDI(populationcontrast_mip)
# contrasts (pop 1=Arizona, pop 2=California)
#  more likely to have relatives in Arizona 
# -0.02921494 -0.01676636  
sum(populationcontrast_mip<0)
# 8000/8000 of posterior samples >0 (100%)
mean(populationcontrast_mip)
# -0.02285802


# More detailed model that takes into account sex

set.seed(1109)
mirmodel <- ulam( alist(
  relative ~dbinom(1,p),
  logit(p) <- pop[population]+s[sex,population],
  pop[population]~dnorm(-5,3),
  vector[2]:s[sex] ~ multi_normal(0,Rho_s,sigma_s),
  Rho_s~dlkjcorr(4),
  sigma_s~dexp(1)
) , data=dat_distantrelatives, chains=4 , cores=4 , cmdstan = TRUE, iter=4000)

precis(mirmodel,depth=3)

# contrasts (pop 1=Arizona, pop 2=California; s 1=FF, s2=FM, s3=MM)
posterior_mir<-extract.samples(mirmodel)
populationcontrast_mir<-inv_logit(posterior_mir$pop[,2])-inv_logit(posterior_mir$pop[,1])
precis(populationcontrast_mir)
HPDI(populationcontrast_mir)

females_populationcontrast_mir<-inv_logit(posterior_mir$s[,1,2]+posterior_mir$pop[,2])-inv_logit(posterior_mir$s[,1,1]+posterior_mir$pop[,1])
precis(females_populationcontrast_mir)
HPDI(females_populationcontrast_mir)


males_populationcontrast_mir<-inv_logit(posterior_mir$s[,3,2]+posterior_mir$pop[,2])-inv_logit(posterior_mir$s[,3,1]+posterior_mir$pop[,1])
precis(males_populationcontrast_mir)
HPDI(males_populationcontrast_mir)
# no difference in the likelihood that male-male dyads are closely related
#                              mean   sd  5.5% 94.5%       histogram
# males_populationcontrast_mir -0.01 0.01 -0.02     0 ▁▁▁▃▇▁▁▁▁▁▁▁
# -0.017703086  0.002763918

opp_populationcontrast_mir<-inv_logit(posterior_mir$s[,2,2]+posterior_mir$pop[,2])-inv_logit(posterior_mir$s[,2,1]+posterior_mir$pop[,1])
precis(opp_populationcontrast_mir)
HPDI(opp_populationcontrast_mir)
# less likely to have opposite sex relatives in California
#                            mean   sd  5.5% 94.5%  histogram
# opp_populationcontrast_mir -0.01  0 -0.01     0 ▁▂▇▂▁▁▁▁▁
# -0.012964160 -0.002121524






```




```{r relatedness permutation cross populations, include=FALSE, echo=FALSE, eval=FALSE}


# Do the cross-population permutation - if we draw X individuals (females in California) from the sample of females in Arizona and check their dyads, how often do we get no  relative?

females_az<-adultids[adultids$Population=="Tempe" & adultids$Sex =="F",]$BirdID
females_az_relatedness<-az_completerelatedness[az_completerelatedness$ind1.id %in% females_az & az_completerelatedness$ind2.id %in% females_az,]
females_az_relatedness_half<-females_az_relatedness[1:(nrow(females_az_relatedness)/2),]
  
counts<-NA
for(i in 1:10000){
  subsetfemales<-sample(females_az,samplesize[5,3])
  subsetrelatedness<-females_az_relatedness_half[females_az_relatedness_half$ind1.id %in% subsetfemales & females_az_relatedness_half$ind2.id %in% subsetfemales,]$kin_qg
  counts<-c(counts,sum(subsetrelatedness))
}
counts<-counts[2:10001]
sum(counts==0)/10000


males_az<-adultids[adultids$Population=="Tempe" & adultids$Sex =="M",]$BirdID
males_az_relatedness_half<-males_az_relatedness[1:(nrow(males_az_relatedness)/2),]
  
counts<-NA
for(i in 1:10000){
  subsetmales<-sample(males_az,samplesize[6,3])
  subsetrelatedness<-males_az_relatedness_half[males_az_relatedness_half$ind1.id %in% subsetmales & males_az_relatedness_half$ind2.id %in% subsetmales,]$kin_qg
  counts<-c(counts,sum(subsetrelatedness))
}
counts<-counts[2:10001]
sum(counts==0)/10000

```



![](gxpop_q2dispersal_figure1.png){width=100%}

Figure 1: The relatedness and geographic distance observed among grackles in Arizona (left, purple) and in California (right, green). Average relatedness is higher in Arizona than in California. In Arizona, there are several pairs of females (lighter circles) and males (darker triangles) who are related at levels higher than 0.25 (dotted line, close kin of half-sibling or closer) and 0.125 (dashed line, distant kin of cousin or closer), while there are no close or distant same-sex kin in California. In terms of potential sex-biases in dispersal, in Arizona, we observe more females than males related at levels of distant kin and of close kin. Closely related females tend to be found at shorter geographic distances than the average female pair, while closely related males are not found at short geographic distances.




#### Sex biases in dispersal in the two populations
*Average relatedness within the sexes*\
Average relatedness among both the females and the males in Arizona is not different from what would be expected by chance. Randomly drawing the same number of individuals from the full sample gives an average relatedness that is lower than that observed among the females in 45% of permutations and for males in 29% of permutations. In contrast, the observed average relatedness among females in California is slightly higher than what would be expected by chance, with 90% of the permutations drawing the same number of individuals from the overall population as there are females leading to lower average relatedness than that observed among the females (Figure 2). In contrast, the observed average relatedness among males is slightly less than what would be expected by chance given the relatedness among individuals in this population, with 91% of permutations giving higher levels of average relatedness than that observed among the males (Figure 2). 

![](gxpop_q2dispersal_figure2.png){width=60%}

Figure 2: In California, the average relatedness observed among females (light green line, left) is higher than the average relatedness observed in most permutations, while the average relatedness observed among males (dark green line, right) is lower than the relatedness observed in most permutations. 10,000 permutations were performed for each sex, drawing a subset of individuals from the total sample and calculating their average relatedness. The width of the violin plot (grey shaded area) reflects the number of permutations during which a particular level of average relatedness was observed.

*Distances among kin of the different sexes*\
In Arizona, the only population where our relatedness calculations indicated likely kin, females related at r≧0.25 are found a median of 391m from each other, while males related at this level are found 1,177m from each other (Figure 1). Similarly, females related at r≧0.125 are found a median of 435m from each other, while males related at this level are found 846m from each other. These differences in distance are not due to females generally being found closer to each other, because in only 2% of permutations drawing from the same number of female and male dyads as those that are related at the respective levels, are the differences in distance as large or larger than those observed (Figure 3). We cannot perform this analysis for California because there are no kin among either sex (Figure 1).

![](gxpop_q2dispersal_figure3.png){width=70%}

Figure 3: In Arizona, females related at r≧0.25 (light purple line, top) are found closer to each other than the average set of females (grey shaded area, top), while males related at r≧0.25 (dark purple line, bottom) are found at larger distances from each other than the average set of males (grey shaded area, bottom). 10,000 permutations were performed for each sex, drawing a subset of individuals from the total sample matching the number of close kin and calculating their average geographic distance The width of the violin plot (grey shaded area) reflects the number of permutations during which a particular average distance was observed.

*Spatial autocorrelation between geographic distance and relatedness in the two sexes*\
The spatial autocorrelation analyses indicate that, in both Arizona and California, female relatives likely stay close to each other while male relatives move away from each other (Figure 1). For Arizona, more closely related females are found at shorter distances from each other (negative correlation between relatedness and geographic distance based on a Mantel test when females are close, -0.08, p=0.02; positive correlation when females are far, 0.06, p=0.03). In contrast, at short distances males are not related to each other (0.05, p=0.21), but relatives are found at larger distances (-0.11, p=0.02). The same pattern is found for females in California, though with the smaller sample size, the effects are not significant (females close -0.15, p=0.12, distant 0.15, p=0.12), whereas for males there is no obvious pattern (close 0.04, p=0.38; distant 0.04, p=0.43). These results are similar when using the larger number of distance classes, with correlations switching from negative to positive for females as distance increases, and from positive to negative for males.


```{r permutation sex bias, include=FALSE, echo=FALSE, eval=FALSE}
#Perform a simulation to assess whether average relatedness among males is different from what we would expect in a random subset of the same number of individuals

# Arizona

#Based on the relatedness estimates based on the method by Queller & Goodnight
#Perform a simulation to assess whether average relatedness among males is different from what we would expect in a random subset of the same number of individuals
simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Tempe",]$BirdID,samplesize[4,3])
  simulatedrelatedness[i,1]<-mean(filter(az_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$quellergt)
}
hist(simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_malerelatednessquellergt_AZ<-sum(simulatedrelatedness>mean(filter(az_pairwise_r,ind1.id %in% males,ind2.id %in% males)$quellergt))/10000
# 0.2893

#Perform a simulation to assess whether average relatedness among females is different from what we would expect in a random subset of the same number of individuals
simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Tempe",]$BirdID,samplesize[3,3])
  simulatedrelatedness[i,1]<-mean(filter(az_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$quellergt)
}
hist(simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_femalerelatednessquellergt_AZ<-sum(simulatedrelatedness>mean(filter(az_pairwise_r,ind1.id %in% females,ind2.id %in% females)$quellergt))/10000
# 0.4481



# California

#Based on the relatedness estimates based on the method by Queller & Goodnight
#Perform a simulation to assess whether average relatedness among males is different from what we would expect in a random subset of the same number of individuals
c_m_simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Woodland",]$BirdID,samplesize[6,3])
  c_m_simulatedrelatedness[i,1]<-mean(filter(ca_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$quellergt)
}
hist(c_m_simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_malerelatednessquellergt_CA<-sum(c_m_simulatedrelatedness>mean(filter(ca_pairwise_r,ind1.id %in% males,ind2.id %in% males)$quellergt))/10000
# 0.9094
# Males slightly less related than expected by chance - 91% of simulated values are larger


#Perform a simulation to assess whether average relatedness among females is different from what we would expect in a random subset of the same number of individuals
c_f_simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Woodland",]$BirdID,samplesize[5,3])
  c_f_simulatedrelatedness[i,1]<-mean(filter(ca_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$quellergt)
}
hist(c_f_simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_femalerelatednessquellergt_CA<-sum(c_f_simulatedrelatedness>mean(filter(ca_pairwise_r,ind1.id %in% females,ind2.id %in% females)$quellergt))/10000
# 0.0973
# Females slightly more related than expected by chance, 90% of simulated values are smaller





simulatedvalues<-as.data.frame(matrix(ncol=2,nrow=20000))
colnames(simulatedvalues)<-c("Sex","Relatedness")
simulatedvalues[1:10000,]$Sex<-"1"
simulatedvalues[1:10000,]$Relatedness<-c_f_simulatedrelatedness
simulatedvalues[10001:20000,]$Sex<-"2"
simulatedvalues[10001:20000,]$Relatedness<-c_m_simulatedrelatedness

library(ggplot2)
g<-ggplot(simulatedvalues,aes(Sex,Relatedness))
g+geom_violin(scale="area",fill="grey90")+labs(y="Average relatedness")+ geom_segment(x = 0.6, y = mean(filter(ca_pairwise_r,ind1.id %in% females,ind2.id %in% females)$quellergt),xend=1.4,yend=mean(filter(ca_pairwise_r,ind1.id %in% females,ind2.id %in% females)$quellergt),col="#ADD4A0",lwd=3)+ geom_segment(x = 1.6, y = mean(filter(ca_pairwise_r,ind1.id %in% males,ind2.id %in% males)$quellergt),xend=2.4,yend=mean(filter(ca_pairwise_r,ind1.id %in% males,ind2.id %in% males)$quellergt),col="#1B7939",lwd=3)+scale_x_discrete(labels = c("1" = "females","2" = "males"))+theme_classic(base_size = 16)+annotate("text", x=0.65, y=-0.02, label= "observed",col="#ADD4A0",size=6)+annotate("text", x=1.65, y=-0.052, label= "observed",col="#1B7939",size=6)+annotate("text", x=1, y=-0.035, label= "permuted",col="grey40",size=6)+annotate("text", x=2, y=-0.035, label= "permuted",col="grey40",size=6)



```








```{r permutation kinship among sexes, eval=FALSE, warning=FALSE, results='asis', echo=FALSE}
options(width=60)

#Analysis 2: Assess whether distances among closely related females are shorter than distances among closely related males


# Arizona
# Select close relatives as pairs of individuals who are related at a level of 0.25 of higher using the Queller & Goodnight estimator
# Among the females
az_close_relatives_females<-filter(az_pairwise_r,quellergt >0.2499,ind1.id %in% females,ind2.id %in% females)
az_close_relatives_females_individuals<-c(az_close_relatives_females$ind1.id,az_close_relatives_females$ind2.id)

#Next subset the the distance matrix to only include these individuals

az_close_relatives_females_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_females),ncol=1)

for (i in 1:nrow(az_close_relatives_females)) {
  ind1<-az_close_relatives_females[i,]$ind1.id
  ind2<-az_close_relatives_females[i,]$ind2.id
  pair_distance<-AZ_female_pairwise_distances[ind1,ind2]
  az_close_relatives_females_pairwise_distances[i,]<-pair_distance
}

median_az_f_025_distances<-median(az_close_relatives_females_pairwise_distances)

hist(az_close_relatives_females_pairwise_distances)


#repeat the same for the males
az_close_relatives_males<-filter(az_pairwise_r,quellergt >0.2499,ind1.id %in% males,ind2.id %in% males)
az_close_relatives_males_individuals<-c(az_close_relatives_males$ind1.id,az_close_relatives_males$ind2.id)

#Next subset the the distance matrix to only include these individuals

az_close_relatives_males_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_males),ncol=1)

for (i in 1:nrow(az_close_relatives_males)) {
  ind1<-az_close_relatives_males[i,]$ind1.id
  ind2<-az_close_relatives_males[i,]$ind2.id
  pair_distance<-AZ_male_pairwise_distances[ind1,ind2]
  az_close_relatives_males_pairwise_distances[i,]<-pair_distance
}

median_az_m_025_distances<-median(az_close_relatives_males_pairwise_distances)

hist(az_close_relatives_males_pairwise_distances)

#calculate difference between the distances among males and among females
observeddifferenceindistances<-median(az_close_relatives_males_pairwise_distances,na.rm=T)-median(az_close_relatives_females_pairwise_distances,na.rm=T)

median_az_difference_025_distances<-observeddifferenceindistances

#perform permutations to generate random draws of matching numbers of individuals to assess whether the sex-difference in the distance is more or less than what would be expected by chance
number_close_relatives_females<-nrow(az_close_relatives_females)
number_close_relatives_males<-nrow(az_close_relatives_males)

simulateddifferencesindistances<-matrix(ncol=1,nrow=10000)
simulateddfemaleindistances<-matrix(ncol=1,nrow=10000)
simulateddmaleindistances<-matrix(ncol=1,nrow=10000)

rownames(AZ_all_pairwise_distance)<-adultids[adultids$Name %in% rownames(AZ_all_pairwise_distance),]$BirdID
colnames(AZ_all_pairwise_distance)<-adultids[adultids$Name %in% colnames(AZ_all_pairwise_distance),]$BirdID

for (i in 1:10000) {
simulated_close_relatives_females<-sample_n(az_pairwise_r, number_close_relatives_females, replace = TRUE)

subset_relatives_females_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_females),ncol=1)

for (j in 1:nrow(simulated_close_relatives_females)) {
  ind1<-simulated_close_relatives_females[j,]$ind1.id
  ind2<-simulated_close_relatives_females[j,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_females_pairwise_distances[j,]<-pair_distance
}

simulated_close_relatives_males<-sample_n(az_pairwise_r, number_close_relatives_males, replace = TRUE)

subset_relatives_males_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_males),ncol=1)

for (k in 1:nrow(simulated_close_relatives_males)) {
  ind1<-simulated_close_relatives_males[k,]$ind1.id
  ind2<-simulated_close_relatives_males[k,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_males_pairwise_distances[k,]<-pair_distance
}

simulateddfemaleindistances[i,1]<-median(subset_relatives_females_pairwise_distances,na.rm=T)
simulateddmaleindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)
simulateddifferencesindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)-median(subset_relatives_females_pairwise_distances,na.rm=T)
  }

simulation_az_f_025_distances<-sum(simulateddfemaleindistances<median(az_close_relatives_females_pairwise_distances))/10000
simulation_az_m_025_distances<-sum(simulateddmaleindistances>median(az_close_relatives_males_pairwise_distances))/10000
simulation_az_difference_025_distances<-sum(simulateddifferencesindistances>observeddifferenceindistances)/10000



# plot the permutation results for the distance among California close kin
simulatedvalues<-as.data.frame(matrix(ncol=2,nrow=20000))
colnames(simulatedvalues)<-c("Sex","Distance")
simulatedvalues[1:10000,]$Sex<-"2"
simulatedvalues[1:10000,]$Distance<-simulateddfemaleindistances
simulatedvalues[10001:20000,]$Sex<-"1"
simulatedvalues[10001:20000,]$Distance<-simulateddmaleindistances

library(ggplot2)
g<-ggplot(simulatedvalues,aes(Sex,Distance))
g+geom_violin(scale="area",fill="grey90")+labs(y="Geographic distance")+ geom_segment(x = 1.6, y = median(az_close_relatives_females_pairwise_distances),xend=2.4,yend=median(az_close_relatives_females_pairwise_distances),col="#C3A4CF",lwd=3)+ geom_segment(x = 0.6, y = median(az_close_relatives_males_pairwise_distances),xend=1.4,yend=median(az_close_relatives_males_pairwise_distances),col="#742881",lwd=3)+scale_x_discrete(labels = c("1" = "males","2" = "females"))+theme_classic(base_size = 16)+annotate("text", x=1.65, y=200, label= "observed",col="#C3A4CF",size=6)+annotate("text", x=0.65, y=1400, label= "observed",col="#742881",size=6)+annotate("text", x=1, y=700, label= "permuted",col="grey40",size=6)+annotate("text", x=2, y=700, label= "permuted",col="grey40",size=6)+coord_flip()





# Arizona
# We also define close relatives as all pairs of individuals who are related by a level of 0.125 or higher (cousins or higher) using the Queller & Goodnight estimator
az_close_relatives_females<-filter(az_pairwise_r,quellergt >0.12499,ind1.id %in% females,ind2.id %in% females)
az_close_relatives_females_individuals<-c(az_close_relatives_females$ind1.id,az_close_relatives_females$ind2.id)

#Next subset the the distance matrix to only include these individuals

az_close_relatives_females_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_females),ncol=1)

for (i in 1:nrow(az_close_relatives_females)) {
  ind1<-az_close_relatives_females[i,]$ind1.id
  ind2<-az_close_relatives_females[i,]$ind2.id
  pair_distance<-AZ_female_pairwise_distances[ind1,ind2]
  az_close_relatives_females_pairwise_distances[i,]<-pair_distance
}

median_az_f_0125_distances<-median(az_close_relatives_females_pairwise_distances)

hist(az_close_relatives_females_pairwise_distances)


#repeat the same for the males
az_close_relatives_males<-filter(az_pairwise_r,quellergt >0.12499,ind1.id %in% males,ind2.id %in% males)
az_close_relatives_males_individuals<-c(az_close_relatives_males$ind1.id,az_close_relatives_males$ind2.id)

#Next subset the the distance matrix to only include these individuals

az_close_relatives_males_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_males),ncol=1)

for (i in 1:nrow(az_close_relatives_males)) {
  ind1<-az_close_relatives_males[i,]$ind1.id
  ind2<-az_close_relatives_males[i,]$ind2.id
  pair_distance<-AZ_male_pairwise_distances[ind1,ind2]
  az_close_relatives_males_pairwise_distances[i,]<-pair_distance
}

median_az_m_0125_distances<-median(az_close_relatives_males_pairwise_distances)

hist(az_close_relatives_males_pairwise_distances)

#calculate difference between the distances among males and among females
observeddifferenceindistances<-median(az_close_relatives_males_pairwise_distances,na.rm=T)-median(az_close_relatives_females_pairwise_distances,na.rm=T)

median_az_difference_0125_distances<-observeddifferenceindistances

#perform simulation to generate random draws of matching numbers of individuals to assess whether the sex-difference in the distance is more or less than what would be expected by chance
number_close_relatives_females<-nrow(az_close_relatives_females)
number_close_relatives_males<-nrow(az_close_relatives_males)

simulateddifferencesindistances<-matrix(ncol=1,nrow=10000)
simulateddfemaleindistances<-matrix(ncol=1,nrow=10000)
simulateddmaleindistances<-matrix(ncol=1,nrow=10000)


for (i in 1:10000) {
simulated_close_relatives_females<-sample_n(az_pairwise_r, number_close_relatives_females, replace = TRUE)

subset_relatives_females_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_females),ncol=1)

for (j in 1:nrow(simulated_close_relatives_females)) {
  ind1<-simulated_close_relatives_females[j,]$ind1.id
  ind2<-simulated_close_relatives_females[j,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_females_pairwise_distances[j,]<-pair_distance
}

simulated_close_relatives_males<-sample_n(az_pairwise_r, number_close_relatives_males, replace = TRUE)

subset_relatives_males_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_males),ncol=1)

for (k in 1:nrow(simulated_close_relatives_males)) {
  ind1<-simulated_close_relatives_males[k,]$ind1.id
  ind2<-simulated_close_relatives_males[k,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_males_pairwise_distances[k,]<-pair_distance
}

simulateddfemaleindistances[i,1]<-median(subset_relatives_females_pairwise_distances,na.rm=T)
simulateddmaleindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)
simulateddifferencesindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)-median(subset_relatives_females_pairwise_distances,na.rm=T)
  }

simulation_az_f_0125_distances<-sum(simulateddfemaleindistances<median(az_close_relatives_females_pairwise_distances))/10000
simulation_az_m_0125_distances<-sum(simulateddmaleindistances>median(az_close_relatives_males_pairwise_distances))/10000
simulation_az_difference_0125_distances<-sum(simulateddifferencesindistances>observeddifferenceindistances)/10000




# California
# In our California sample, there is only one opposite-sex pair of individuals who is related at the level of 0.125 or higher. We therefore cannot perform the simulations in the same way. 

```




```{r spatial correlogram for relatedness in the two sexes, eval=FALSE, warning=FALSE, results='asis', echo=FALSE}
options(width=60)

#Analysis 3: Correlogram to assess change of relatedness with distances

library(DataCombine)

# Start with Arizona:

#turn pairwise_r$quellergt into a matrix
az_relatedness_matrix<-select(az_pairwise_r,ind1.id,ind2.id,quellergt)
az_relatedness_matrix<-spread(az_relatedness_matrix,"ind1.id","quellergt")
az_relatedness_matrix <- cbind(az_relatedness_matrix,Add=NA)
colnames(az_relatedness_matrix)<-c(colnames(az_relatedness_matrix)[1:(samplesize[3,3]+samplesize[4,3])],sort(unique(adultids[adultids$Population=="Tempe",]$BirdID))[samplesize[3,3]+samplesize[4,3]])
az_relatedness_matrix<-arrange(az_relatedness_matrix,ind2.id)
az_relatedness_matrix <- InsertRow(data=az_relatedness_matrix, NewRow=rep(NA,samplesize[3,3]+samplesize[4,3]+1), RowNum=1)
az_relatedness_matrix[1,1] <- sort(unique(adultids[adultids$Population=="Tempe",]$BirdID))[1]
rownames(az_relatedness_matrix)<-az_relatedness_matrix[,1]
az_relatedness_matrix<-az_relatedness_matrix[1:nrow(az_relatedness_matrix),2:ncol(az_relatedness_matrix)]




az_female_relatedness_matrix<-az_relatedness_matrix[rownames(az_relatedness_matrix) %in% females,colnames(az_relatedness_matrix) %in% females]
az_male_relatedness_matrix<-az_relatedness_matrix[rownames(az_relatedness_matrix) %in% males,colnames(az_relatedness_matrix) %in% males]

AZ_female_pairwise_distances_r<-AZ_female_pairwise_distances[order(factor(rownames(AZ_female_pairwise_distances), levels = rownames(az_female_relatedness_matrix), ordered=TRUE)),order(factor(colnames(AZ_female_pairwise_distances), levels = colnames(az_female_relatedness_matrix), ordered=TRUE))]



# for (i in 1:ncol(AZ_female_pairwise_distances_r)) {
#  AZ_female_pairwise_distances_r[i,i:ncol(AZ_female_pairwise_distances_r)]<-NA
# }

AZ_male_pairwise_distances_r<-AZ_male_pairwise_distances[order(factor(rownames(AZ_male_pairwise_distances), levels = rownames(az_male_relatedness_matrix), ordered=TRUE)),order(factor(colnames(AZ_male_pairwise_distances), levels = colnames(az_male_relatedness_matrix), ordered=TRUE))]

AZ_male_pairwise_distances_r<-AZ_male_pairwise_distances_r[rownames(AZ_male_pairwise_distances_r) %in% rownames(az_male_relatedness_matrix),colnames(AZ_male_pairwise_distances_r) %in% colnames(az_male_relatedness_matrix)]

for (i in 1:ncol(AZ_male_pairwise_distances_r)) {
  AZ_male_pairwise_distances_r[i,i:ncol(AZ_male_pairwise_distances_r)]<-NA
}


az_female_relatedness_matrix[, 1:ncol(az_female_relatedness_matrix)] <- apply(az_female_relatedness_matrix[, 1:ncol(az_female_relatedness_matrix)], 2, function(x) as.numeric(as.character(x)))

az_male_relatedness_matrix[, 1:ncol(az_male_relatedness_matrix)] <- apply(az_male_relatedness_matrix[, 1:ncol(az_male_relatedness_matrix)], 2, function(x) as.numeric(as.character(x)))

AZ_female_pairwise_distances_r[, 1:ncol(AZ_female_pairwise_distances_r)] <- apply(AZ_female_pairwise_distances_r[, 1:ncol(AZ_female_pairwise_distances_r)], 2, function(x) as.numeric(as.character(x)))

rownames(AZ_female_pairwise_distances_r)<-gsub("-","_",rownames(AZ_female_pairwise_distances_r))
colnames(AZ_female_pairwise_distances_r)<-gsub("-","_",colnames(AZ_female_pairwise_distances_r))
rownames(az_female_relatedness_matrix)<-gsub("-","_",rownames(az_female_relatedness_matrix))
colnames(az_female_relatedness_matrix)<-gsub("-","_",colnames(az_female_relatedness_matrix))

rownames(AZ_female_pairwise_distances_r)<-gsub("A","AF_",rownames(AZ_female_pairwise_distances_r))
colnames(AZ_female_pairwise_distances_r)<-gsub("A","AF_",colnames(AZ_female_pairwise_distances_r))
rownames(az_female_relatedness_matrix)<-gsub("A","AF_",rownames(az_female_relatedness_matrix))
colnames(az_female_relatedness_matrix)<-gsub("A","AF_",colnames(az_female_relatedness_matrix))

#perform the correlogram analysis
#first way, defining the distance classes
az_female_correlogram_setdistances<-mantel.correlog(D.eco=az_female_relatedness_matrix,D.geo=AZ_female_pairwise_distances_r,break.pts=c(0,100,200,300,400,500,750,1000,1250,2000),cutoff=FALSE,nperm=10000)
az_male_correlogram_setdistances<-mantel.correlog(D.eco=az_male_relatedness_matrix,D.geo=AZ_male_pairwise_distances,break.pts=c(0,100,200,300,400,500,750,1000,1250,2000),cutoff=FALSE,nperm=10000)
#second way, setting the number of distance classes
az_female_correlogram_classes<-mantel.correlog(D.eco=az_female_relatedness_matrix,D.geo=AZ_female_pairwise_distances_r,n.class=5)
az_male_correlogram_classes<-mantel.correlog(D.eco=az_male_relatedness_matrix,D.geo=AZ_male_pairwise_distances,n.class=5)

#additional way, with the distance classes based on the inferred distance among relatives from analysis ii
female_correlogram_setdistances_adj<-mantel.correlog(D.eco=az_female_relatedness_matrix,D.geo=AZ_female_pairwise_distances_r,break.pts=c(0,350,700,1050,1400),cutoff=FALSE,nperm=10000)
male_correlogram_setdistances_adj<-mantel.correlog(D.eco=az_male_relatedness_matrix,D.geo=AZ_male_pairwise_distances,break.pts=c(0,350,700,1050,1400),cutoff=FALSE,nperm=10000)

female_correlogram_setdistances_adj
#       class.index      n.dist  Mantel.cor Pr(Mantel) Pr(corrected)  
# D.cl.1  1.7500e+02  5.6400e+02 -4.9306e-02     0.0720       0.07199 .
# D.cl.2  5.2500e+02  9.3200e+02 -1.4069e-02     0.3601       0.36006  
# D.cl.3  8.7500e+02  7.6400e+02  2.8442e-03     0.4770       0.72013  
# D.cl.4  1.2250e+03  2.2600e+02  6.4680e-02     0.0347       0.13879 


male_correlogram_setdistances_adj

#        class.index      n.dist  Mantel.cor Pr(Mantel) Pr(corrected)  
# D.cl.1  175.000000  120.000000    0.053126     0.1928       0.19278  
# D.cl.2  525.000000  190.000000   -0.113895     0.0370       0.07399 .
# D.cl.3  875.000000  174.000000   -0.112598     0.0276       0.08279 .
# D.cl.4 1225.000000  142.000000    0.126563     0.0142       0.05679 .


# Simpler, split it into two categories: close (within 400m) vs far away (more than 400m)
# females
#        class.index      n.dist  Mantel.cor Pr(Mantel) Pr(corrected)  
# D.cl.1  200.000000  670.000000   -0.078201     0.0153        0.0153 *
# D.cl.2  900.000000 1816.000000    0.056897     0.0267        0.0306 *
# males
# D.cl.1  200.000000 152.000000   0.050486     0.2131        0.2131  
# D.cl.2  900.000000 474.000000  -0.105008     0.0188        0.0376 *



# Next California:

#turn pairwise_r$quellergt into a matrix
ca_relatedness_matrix<-select(ca_pairwise_r,ind1.id,ind2.id,quellergt)
ca_relatedness_matrix<-spread(ca_relatedness_matrix,"ind1.id","quellergt")
ca_relatedness_matrix <- cbind(ca_relatedness_matrix,Add=NA)
colnames(ca_relatedness_matrix)<-c(colnames(ca_relatedness_matrix)[1:(samplesize[5,3]+samplesize[6,3])],sort(unique(adultids[adultids$Population=="Woodland",]$BirdID))[samplesize[5,3]+samplesize[6,3]])
ca_relatedness_matrix<-arrange(ca_relatedness_matrix,ind2.id)
ca_relatedness_matrix <- InsertRow(data=ca_relatedness_matrix, NewRow=rep(NA,samplesize[5,3]+samplesize[6,3]+1), RowNum=1)
ca_relatedness_matrix[1,1] <- sort(unique(adultids[adultids$Population=="Woodland",]$BirdID))[1]
rownames(ca_relatedness_matrix)<-ca_relatedness_matrix[,1]
ca_relatedness_matrix<-ca_relatedness_matrix[1:nrow(ca_relatedness_matrix),2:ncol(ca_relatedness_matrix)]

ca_female_relatedness_matrix<-ca_relatedness_matrix[rownames(ca_relatedness_matrix) %in% females,colnames(ca_relatedness_matrix) %in% females]
ca_male_relatedness_matrix<-ca_relatedness_matrix[rownames(ca_relatedness_matrix) %in% males,colnames(ca_relatedness_matrix) %in% males]

ca_female_pairwise_distances_r<-CA_female_pairwise_distances[order(factor(rownames(CA_female_pairwise_distances), levels = rownames(ca_female_relatedness_matrix), ordered=TRUE)),order(factor(colnames(CA_female_pairwise_distances), levels = colnames(ca_female_relatedness_matrix), ordered=TRUE))]



# for (i in 1:ncol(ca_female_pairwise_distances_r)) {
#  ca_female_pairwise_distances_r[i,i:ncol(ca_female_pairwise_distances_r)]<-NA
# }

ca_male_pairwise_distances_r<-CA_male_pairwise_distances[order(factor(rownames(CA_male_pairwise_distances), levels = rownames(ca_male_relatedness_matrix), ordered=TRUE)),order(factor(colnames(CA_male_pairwise_distances), levels = colnames(ca_male_relatedness_matrix), ordered=TRUE))]


for (i in 1:ncol(ca_male_pairwise_distances_r)) {
  ca_male_pairwise_distances_r[i,i:ncol(ca_male_pairwise_distances_r)]<-NA
}


ca_female_relatedness_matrix[, 1:ncol(ca_female_relatedness_matrix)] <- apply(ca_female_relatedness_matrix[, 1:ncol(ca_female_relatedness_matrix)], 2, function(x) as.numeric(as.character(x)))

ca_male_relatedness_matrix[, 1:ncol(ca_male_relatedness_matrix)] <- apply(ca_male_relatedness_matrix[, 1:ncol(ca_male_relatedness_matrix)], 2, function(x) as.numeric(as.character(x)))

ca_female_pairwise_distances_r[, 1:ncol(ca_female_pairwise_distances_r)] <- apply(ca_female_pairwise_distances_r[, 1:ncol(ca_female_pairwise_distances_r)], 2, function(x) as.numeric(as.character(x)))

rownames(ca_female_pairwise_distances_r)<-gsub("-","_",rownames(ca_female_pairwise_distances_r))
colnames(ca_female_pairwise_distances_r)<-gsub("-","_",colnames(ca_female_pairwise_distances_r))
rownames(ca_female_relatedness_matrix)<-gsub("-","_",rownames(ca_female_relatedness_matrix))
colnames(ca_female_relatedness_matrix)<-gsub("-","_",colnames(ca_female_relatedness_matrix))

rownames(ca_female_pairwise_distances_r)<-gsub("A","AF_",rownames(ca_female_pairwise_distances_r))
colnames(ca_female_pairwise_distances_r)<-gsub("A","AF_",colnames(ca_female_pairwise_distances_r))
rownames(ca_female_relatedness_matrix)<-gsub("A","AF_",rownames(ca_female_relatedness_matrix))
colnames(ca_female_relatedness_matrix)<-gsub("A","AF_",colnames(ca_female_relatedness_matrix))

#perform the correlogram analysis
#first way, defining the distance classes
ca_female_correlogram_setdistances<-mantel.correlog(D.eco=ca_female_relatedness_matrix,D.geo=ca_female_pairwise_distances_r,break.pts=c(0,100,200,300,400,500,750,1000,1250,2000),cutoff=FALSE,nperm=10000)
ca_male_correlogram_setdistances<-mantel.correlog(D.eco=ca_male_relatedness_matrix,D.geo=ca_male_pairwise_distances_r,break.pts=c(0,100,200,300,400,500,750,1000,1250,2000),cutoff=FALSE,nperm=10000)
#second way, setting the number of distance classes
ca_female_correlogram_classes<-mantel.correlog(D.eco=ca_female_relatedness_matrix,D.geo=ca_female_pairwise_distances_r,n.class=5)
ca_male_correlogram_classes<-mantel.correlog(D.eco=ca_male_relatedness_matrix,D.geo=ca_male_pairwise_distances_r,n.class=5)

#additional way, with the distance classes based on the inferred distance among relatives from analysis ii
ca_female_correlogram_setdistances_adj<-mantel.correlog(D.eco=ca_female_relatedness_matrix,D.geo=ca_female_pairwise_distances_r,break.pts=c(0,350,700,1050,1400,4100,8200,35000),cutoff=FALSE,nperm=10000)
ca_male_correlogram_setdistances_adj<-mantel.correlog(D.eco=ca_male_relatedness_matrix,D.geo=ca_male_pairwise_distances_r,break.pts=c(0,350,700,1050,1400,4100,8200,35000),cutoff=FALSE,nperm=10000)

ca_female_correlogram_setdistances_adj
ca_male_correlogram_setdistances_adj




# California with only the birds from a single location
#turn pairwise_r$quellergt into a matrix
ca_relatedness_matrix<-select(ca_pairwise_r,ind1.id,ind2.id,quellergt)
ca_relatedness_matrix<-spread(ca_relatedness_matrix,"ind1.id","quellergt")
ca_relatedness_matrix <- cbind(ca_relatedness_matrix,Add=NA)
colnames(ca_relatedness_matrix)<-c(colnames(ca_relatedness_matrix)[1:(samplesize[5,3]+samplesize[6,3])],sort(unique(adultids[adultids$Population=="Woodland",]$BirdID))[samplesize[5,3]+samplesize[6,3]])
ca_relatedness_matrix<-arrange(ca_relatedness_matrix,ind2.id)
ca_relatedness_matrix <- InsertRow(data=ca_relatedness_matrix, NewRow=rep(NA,samplesize[5,3]+samplesize[6,3]+1), RowNum=1)
ca_relatedness_matrix[1,1] <- sort(unique(adultids[adultids$Population=="Woodland",]$BirdID))[1]
rownames(ca_relatedness_matrix)<-ca_relatedness_matrix[,1]
ca_relatedness_matrix<-ca_relatedness_matrix[1:nrow(ca_relatedness_matrix),2:ncol(ca_relatedness_matrix)]

ca_female_relatedness_matrix<-ca_relatedness_matrix[rownames(ca_relatedness_matrix) %in% females,colnames(ca_relatedness_matrix) %in% females]
ca_male_relatedness_matrix<-ca_relatedness_matrix[rownames(ca_relatedness_matrix) %in% males,colnames(ca_relatedness_matrix) %in% males]

ca_female_pairwise_distances_r<-CA_female_pairwise_distances_singlelocation[order(factor(rownames(CA_female_pairwise_distances_singlelocation), levels = rownames(ca_female_relatedness_matrix), ordered=TRUE)),order(factor(colnames(CA_female_pairwise_distances_singlelocation), levels = colnames(ca_female_relatedness_matrix), ordered=TRUE))]

ca_female_relatedness_matrix<-ca_female_relatedness_matrix[rownames(ca_female_relatedness_matrix) %in% rownames(ca_female_pairwise_distances_r),colnames(ca_female_relatedness_matrix) %in% colnames(ca_female_pairwise_distances_r)]

ca_male_pairwise_distances_r<-CA_male_pairwise_distances[order(factor(rownames(CA_male_pairwise_distances_singlelocation), levels = rownames(ca_male_relatedness_matrix), ordered=TRUE)),order(factor(colnames(CA_male_pairwise_distances_singlelocation), levels = colnames(ca_male_relatedness_matrix), ordered=TRUE))]

ca_male_relatedness_matrix<-ca_male_relatedness_matrix[rownames(ca_male_relatedness_matrix) %in% rownames(ca_male_pairwise_distances_r),colnames(ca_male_relatedness_matrix) %in% colnames(ca_male_pairwise_distances_r)]

for (i in 1:ncol(ca_male_pairwise_distances_r)) {
  ca_male_pairwise_distances_r[i,i:ncol(ca_male_pairwise_distances_r)]<-NA
}


ca_female_relatedness_matrix[, 1:ncol(ca_female_relatedness_matrix)] <- apply(ca_female_relatedness_matrix[, 1:ncol(ca_female_relatedness_matrix)], 2, function(x) as.numeric(as.character(x)))

ca_male_relatedness_matrix[, 1:ncol(ca_male_relatedness_matrix)] <- apply(ca_male_relatedness_matrix[, 1:ncol(ca_male_relatedness_matrix)], 2, function(x) as.numeric(as.character(x)))

ca_female_pairwise_distances_r[, 1:ncol(ca_female_pairwise_distances_r)] <- apply(ca_female_pairwise_distances_r[, 1:ncol(ca_female_pairwise_distances_r)], 2, function(x) as.numeric(as.character(x)))

rownames(ca_female_pairwise_distances_r)<-gsub("-","_",rownames(ca_female_pairwise_distances_r))
colnames(ca_female_pairwise_distances_r)<-gsub("-","_",colnames(ca_female_pairwise_distances_r))
rownames(ca_female_relatedness_matrix)<-gsub("-","_",rownames(ca_female_relatedness_matrix))
colnames(ca_female_relatedness_matrix)<-gsub("-","_",colnames(ca_female_relatedness_matrix))

rownames(ca_female_pairwise_distances_r)<-gsub("A","AF_",rownames(ca_female_pairwise_distances_r))
colnames(ca_female_pairwise_distances_r)<-gsub("A","AF_",colnames(ca_female_pairwise_distances_r))
rownames(ca_female_relatedness_matrix)<-gsub("A","AF_",rownames(ca_female_relatedness_matrix))
colnames(ca_female_relatedness_matrix)<-gsub("A","AF_",colnames(ca_female_relatedness_matrix))

#perform the correlogram analysis
#first way, defining the distance classes
ca_female_correlogram_setdistances<-mantel.correlog(D.eco=ca_female_relatedness_matrix,D.geo=ca_female_pairwise_distances_r,break.pts=c(0,100,200,300,400,500,750,1000,1250,2000),cutoff=FALSE,nperm=10000)
ca_male_correlogram_setdistances<-mantel.correlog(D.eco=ca_male_relatedness_matrix,D.geo=ca_male_pairwise_distances_r,break.pts=c(0,100,200,300,400,500,750,1000,1250,2000),cutoff=FALSE,nperm=10000)
#second way, setting the number of distance classes
ca_female_correlogram_classes<-mantel.correlog(D.eco=ca_female_relatedness_matrix,D.geo=ca_female_pairwise_distances_r,n.class=5)
ca_male_correlogram_classes<-mantel.correlog(D.eco=ca_male_relatedness_matrix,D.geo=ca_male_pairwise_distances_r,n.class=5)

#additional way, with the distance classes based on the inferred distance among relatives from analysis ii
ca_female_correlogram_setdistances_adj<-mantel.correlog(D.eco=ca_female_relatedness_matrix,D.geo=ca_female_pairwise_distances_r,break.pts=c(0,350,700,1050,1400),cutoff=FALSE,nperm=10000)
ca_male_correlogram_setdistances_adj<-mantel.correlog(D.eco=ca_male_relatedness_matrix,D.geo=ca_male_pairwise_distances_r,break.pts=c(0,350,700,1050,1400),cutoff=FALSE,nperm=10000)

ca_female_correlogram_setdistances_adj
#        class.index      n.dist  Mantel.cor Pr(Mantel) Pr(corrected)
# D.cl.1  1.7500e+02  4.2000e+01 -4.2917e-02     0.3625        0.3625
# D.cl.2  5.2500e+02  4.8000e+01  1.0090e-02     0.4499        0.7249
# D.cl.3  8.7500e+02  3.0000e+01  4.2440e-02     0.3624        1.0000
# D.cl.4  1.2250e+03  1.2000e+01 -9.2164e-03     0.5326        1.0000

ca_male_correlogram_setdistances_adj
#        class.index      n.dist  Mantel.cor Pr(Mantel) Pr(corrected)
# D.cl.1  175.000000   12.000000    0.039802     0.3783        0.3783
# D.cl.2  525.000000   14.000000    0.035077     0.3927        0.7565
# D.cl.3  875.000000    8.000000    0.010982     0.4529        1.0000
# D.cl.4 1225.000000    2.000000   -0.013693     0.4743        1.0000

# Simpler, close (within 400m) versus far (more than 400m)
# females
#        class.index    n.dist Mantel.cor Pr(Mantel) Pr(corrected)
# D.cl.1   200.00000  54.00000   -0.15343     0.1231        0.1231
# D.cl.2   900.00000  78.00000    0.15343     0.1193        0.2386
# males
# D.cl.1  200.000000  12.000000   0.039802     0.3754        0.3754
# D.cl.2  900.000000  24.000000   0.032382     0.4298        0.7507

```



```{r plot relatedness over distance, eval=FALSE, warning=FALSE, results='asis', echo=FALSE}
options(width=60)

ca_male_relatedness_vector<-paste(as.matrix(ca_male_relatedness_matrix))
ca_male_distance_vector<-paste(ca_male_pairwise_distances_r)
ca_male_distance_vector<-ca_male_distance_vector[ca_male_relatedness_vector!="NA"]
ca_male_relatedness_vector<-ca_male_relatedness_vector[ca_male_relatedness_vector!="NA"]
ca_male_relatedness_vector<-as.numeric(ca_male_relatedness_vector)
ca_male_distance_vector<-as.numeric(ca_male_distance_vector)

az_male_relatedness_vector<-paste(as.matrix(az_male_relatedness_matrix))
az_male_distance_vector<-paste(AZ_male_pairwise_distances_r)
az_male_relatedness_vector<-az_male_relatedness_vector[az_male_relatedness_vector!="NA"]
az_male_distance_vector<-az_male_distance_vector[az_male_distance_vector!="NA"]
az_male_relatedness_vector<-as.numeric(az_male_relatedness_vector)
az_male_distance_vector<-as.numeric(az_male_distance_vector)



ca_female_relatedness_vector<-paste(as.matrix(ca_female_relatedness_matrix))
ca_female_distance_vector<-paste(ca_female_pairwise_distances_r)
ca_female_distance_vector<-ca_female_distance_vector[ca_female_relatedness_vector!="NA"]
ca_female_relatedness_vector<-ca_female_relatedness_vector[ca_female_relatedness_vector!="NA"]
ca_female_relatedness_vector<-as.numeric(ca_female_relatedness_vector)
ca_female_distance_vector<-as.numeric(ca_female_distance_vector)


az_female_relatedness_vector<-paste(as.matrix(az_female_relatedness_matrix))
az_female_distance_vector<-paste(AZ_female_pairwise_distances_r)
az_female_distance_vector<-az_female_distance_vector[az_female_relatedness_vector!="NA"]
az_female_relatedness_vector<-az_female_relatedness_vector[az_female_relatedness_vector!="NA"]
az_female_relatedness_vector<-as.numeric(az_female_relatedness_vector)
az_female_distance_vector<-as.numeric(az_female_distance_vector)


par(mfrow=c(1,2))
plot( NULL , xlim=c(0,2000) , ylim=c(-0.3,0.6) , cex=2,cex.lab=1.5,font=2 ,bty="n",ylab="",xlab="" )
points(az_female_relatedness_vector~az_female_distance_vector,col="#C3A4CF",pch=16,cex=2)
points(az_male_relatedness_vector~az_male_distance_vector,col="#742881",pch=17,cex=2)
legend(x=1500,y=0.7, legend=c(pch16="Females", pch17="Males"), pch=c(16,17), col=c("#C3A4CF","#742881"), box.lty=1, cex=1.2,pt.cex=1.7)
mtext("Relatedness" , side=2,font=2,cex=1.5,line=2.4)
mtext("Geographic distance (m)",side=1,font=2,cex=1.5,line=3)
abline(a=0.125,b=0,lty=2,lwd=1.25,col="grey20")
abline(a=0.25,b=0,lty=3,lwd=1.25,col="grey50")
text("close kin",x=1890,y=0.285)
arrows(x0=2040,x1=2040,y0=0.255,y1=0.335,length=0.1)
text("distant kin",x=1890,y=0.16)
arrows(x0=2040,x1=2040,y0=0.13,y1=0.21,length=0.1)
mtext("Arizona", side=3,line=1.5,font=2,at=800,cex=2,col="#986EAC")

plot( NULL , xlim=c(0,2000) , ylim=c(-0.3,0.6) , cex=2,cex.lab=1.5,font=2 ,bty="n",ylab="",xlab="" )
points(ca_female_relatedness_vector~ca_female_distance_vector,col="#ADD4A0",pch=16,cex=2)
points(ca_male_relatedness_vector~ca_male_distance_vector,col="#1B7939",pch=17,cex=2)
legend(x=1500,y=0.7, legend=c(pch16="Females", pch17="Males"), pch=c(16,17), col=c("#ADD4A0","#1B7939"), box.lty=1, cex=1.2,pt.cex=1.7)
mtext("Relatedness" , side=2,font=2,cex=1.5,line=2.4)
mtext("Geographic distance (m)",side=1,font=2,cex=1.5,line=3)
abline(a=0.125,b=0,lty=2,lwd=1.25,col="grey20")
abline(a=0.25,b=0,lty=3,lwd=1.25,col="grey75")
text("close kin",x=1890,y=0.285)
arrows(x0=2040,x1=2040,y0=0.255,y1=0.335,length=0.1)
text("distant kin",x=1890,y=0.16)
arrows(x0=2040,x1=2040,y0=0.13,y1=0.21,length=0.1)
mtext("California", side=3,line=1.5,font=2,at=800,cex=2,col="#5CAE63")

```








```{r network model cross population, include=FALSE, echo=FALSE, eval=FALSE}

# These are the unregistered analyses presented in the discussion, using a network model to determine whether the number of close relatives is larger in one population then the other

library(STRAND)

# Combine kinship data from both populations to determine whether individuals in Arizona are more likely to be in a kin tie compared to individuals in California


# Convert Arizona data into a matrix
az_relatednessmatrix<-as.data.frame(pivot_wider(az_completerelatedness,id_cols=ind1.id,names_from=ind2.id,values_from=quellergt))

row.names(az_relatednessmatrix)<-az_relatednessmatrix$ind1.id
az_relatednessmatrix<-az_relatednessmatrix[,2:80]
az_relatednessmatrix<-az_relatednessmatrix[,c(79,1:78)]

az_birds<-adultids[adultids$Population %in% "Tempe",]
az_birds<-az_birds[order(az_birds$BirdID),]

az_relatives<-(az_relatednessmatrix>0.1249)*1
diag(az_relatives)<-0

# Convert California data into a matrix
ca_relatednessmatrix<-as.data.frame(pivot_wider(ca_completerelatedness,id_cols=ind1.id,names_from=ind2.id,values_from=quellergt))

row.names(ca_relatednessmatrix)<-ca_relatednessmatrix$ind1.id
ca_relatednessmatrix<-ca_relatednessmatrix[,2:29]
ca_relatednessmatrix<-ca_relatednessmatrix[,c(28,1:27)]

ca_birds<-adultids[adultids$Population %in% "Woodland",]
ca_birds<-ca_birds[order(ca_birds$BirdID),]

ca_relatives<-(ca_relatednessmatrix>0.1249)*1
diag(ca_relatives)<-0

# Combine the two populations
combined_relatives<-as.data.frame(matrix(rep(0,(nrow(az_relatives)+nrow(ca_relatives))^2),ncol=ncol(az_relatives)+ncol(ca_relatives),nrow=nrow(az_relatives)+nrow(ca_relatives)))

combined_relatives[1:nrow(az_relatives),1:ncol(az_relatives)]<-az_relatives
combined_relatives[(nrow(az_relatives+1)):nrow(combined_relatives),(ncol(az_relatives)+1):ncol(combined_relatives)]
colnames(combined_relatives)<-c(colnames(az_relatives),colnames(ca_relatives))
rownames(combined_relatives)<-c(rownames(az_relatives),rownames(ca_relatives))

# We have our outcome variable
outcome=list(Relative=as.matrix(combined_relatives))

# The predictor variable is which of the two population an individual belongs to
group_ids=data.frame(Population=as.factor(c(rep("AZ",nrow(az_relatives)),rep("CA",nrow(ca_relatives)))))
rownames(group_ids)<-rownames(combined_relatives)

# We only estimated kinship within each population, and we therefore cannot make inferences about the kinship relations between populations. We therefore mask all dyads that would be between individuals in different populations
Otherpopulation<-as.data.frame(matrix(rep(0,(nrow(az_relatives)+nrow(ca_relatives))^2),ncol=ncol(az_relatives)+ncol(ca_relatives),nrow=nrow(az_relatives)+nrow(ca_relatives)))

Otherpopulation[(nrow(az_relatives)+1):(nrow(az_relatives)+nrow(ca_relatives)),1:ncol(az_relatives)]<-1
Otherpopulation[1:nrow(az_relatives),(ncol(az_relatives)+1):(ncol(az_relatives)+ncol(ca_relatives))]<-1

rownames(Otherpopulation)<-rownames(combined_relatives)
diag(Otherpopulation)<-1
colnames(Otherpopulation)<-colnames(combined_relatives)

mask=list(Relative=as.matrix(Otherpopulation))

# Create the data frame and define the model
dat = make_strand_data(
  outcome = outcome,
  block_covariates = group_ids ,
  individual_covariates = NULL,
  dyadic_covariates = NULL,
  mask=mask,
  outcome_mode = "bernoulli"
  )


fit =
  fit_block_plus_social_relations_model(
    data=dat,
    block_regression = ~ Population,
    focal_regression = ~ 1,
    dyad_regression = ~ 1,
    target_regression = ~ 1,
    mode="mcmc",
    stan_mcmc_parameters = list(
      chains = 1,
      iter_warmup = 1500,
      iter_sampling = 1500)
    )



# Extract results
res = summarize_strand_results(fit)
res$summary
# m-m dyads slightly more likely to be kin than f-f dyads, and Arizona birds more likely to be kin than California birds


# Calculate contrasts for the model
pop_samps = res$sample$srm_model_samples$block_parameters[[2]]
# The above line of code pulls out the MCMC samples for the within- and between-block tie parameters
# The first slot, [[1]], is for the intercept, and the next slot, [[2]], is for the population variable.

# Check the order of the factor levels
attr(dat,"group_ids_levels")$Population
# "AZ" is in slot 1, and "CA" is in slot 2

# Now, compute the contrast of female-to-female versus male-to-male ties
mean(pop_samps[,1,1] - pop_samps[,2,2])
# 3.783862
HPDI(pop_samps[,1,1] - pop_samps[,2,2], prob=0.89)
#    0.89    0.89
# 0.93967 6.81420  



# Split individuals by their sex

# Obtain the sex of each individual in this dataset
includedbirds<-as.data.frame(rownames(combined_relatives))
colnames(includedbirds)<-"ind2.id"
includedbirds<-merge(includedbirds, birdsex, by="ind2.id")

# subset the data frames to only include females
f_combined_relatives<-combined_relatives[includedbirds$ind2.sex=="F",includedbirds$ind2.sex=="F"]

f_outcome=list(Relative=as.matrix(f_combined_relatives))

f_Otherpopulation<-Otherpopulation[includedbirds$ind2.sex=="F",includedbirds$ind2.sex=="F"]

f_mask=list(Relative=as.matrix(f_Otherpopulation))

f_group_ids<-group_ids[includedbirds$ind2.sex=="F",]

f_group_ids=data.frame(Population=as.factor(f_group_ids))
rownames(f_group_ids)<-rownames(f_combined_relatives)



# Create the data frame and define the model
f_dat = make_strand_data(
  outcome = f_outcome,
  block_covariates = f_group_ids ,
  individual_covariates = NULL,
  dyadic_covariates = NULL,
  mask=f_mask,
  outcome_mode = "bernoulli"
  )

f_fit =
  fit_block_plus_social_relations_model(
    data=f_dat,
    block_regression = ~ Population,
    focal_regression = ~ 1,
    dyad_regression = ~ 1,
    target_regression = ~ 1,
    mode="mcmc",
    stan_mcmc_parameters = list(
      chains = 1,
      iter_warmup = 1500,
      iter_sampling = 1500)
    )


f_res = summarize_strand_results(f_fit)
f_pop_samps = f_res$sample$srm_model_samples$block_parameters[[2]]
# The above line of code pulls out the MCMC samples for the within- and between-block tie parameters
# The first slot, [[1]], is for the intercept, and the next slot, [[2]], is for the population variable

# Check the order of the factor levels
attr(dat,"group_ids_levels")$Population
# "AZ" is in slot 1, and "CA" is in slot 2 \ we want a comparison of the within population ties (i.e. AZ individuals to AZ individuals - 1,1 compared to CA individuals to CA individuals - 2,2)


# Contrast for females
HPDI( f_pop_samps[,1,1] - f_pop_samps[,2,2], prob=0.89)
#    0.89    0.89
# -2.35981  4.78737  





# Obtain the sex of each individual in this dataset
includedbirds<-as.data.frame(rownames(combined_relatives))
colnames(includedbirds)<-"ind2.id"
includedbirds<-merge(includedbirds, birdsex, by="ind2.id")

# subset the data frames to only include females
m_combined_relatives<-combined_relatives[includedbirds$ind2.sex=="M",includedbirds$ind2.sex=="M"]

m_outcome=list(Relative=as.matrix(m_combined_relatives))

m_Otherpopulation<-Otherpopulation[includedbirds$ind2.sex=="M",includedbirds$ind2.sex=="M"]

m_mask=list(Relative=as.matrix(m_Otherpopulation))

m_group_ids<-group_ids[includedbirds$ind2.sex=="M",]

m_group_ids=data.frame(Population=as.factor(m_group_ids))
rownames(m_group_ids)<-rownames(m_combined_relatives)



# Create the data frame and define the model
m_dat = make_strand_data(
  outcome = m_outcome,
  block_covariates = m_group_ids ,
  individual_covariates = NULL,
  dyadic_covariates = NULL,
  mask=m_mask,
  outcome_mode = "bernoulli"
  )

m_fit =
  fit_block_plus_social_relations_model(
    data=m_dat,
    block_regression = ~ Population,
    focal_regression = ~ 1,
    dyad_regression = ~ 1,
    target_regression = ~ 1,
    mode="mcmc",
    stan_mcmc_parameters = list(
      chains = 1,
      iter_warmup = 1500,
      iter_sampling = 1500)
    )


m_res = summarize_strand_results(m_fit)
m_pop_samps = m_res$sample$srm_model_samples$block_parameters[[2]]
# The above line of code pulls out the MCMC samples for the within- and between-block tie parameters
# The first slot, [[1]], is for the intercept, and the next slot, [[2]], is for the population variable

# Check the order of the factor levels
attr(dat,"group_ids_levels")$Population
# "AZ" is in slot 1, and "CA" is in slot 2 \ we want a comparison of the within population ties (i.e. AZ individuals to AZ individuals - 1,1 compared to CA individuals to CA individuals - 2,2)


# Contrast for females
HPDI( m_pop_samps[,1,1] - m_pop_samps[,2,2], prob=0.89)
#    0.89    0.89 
# -0.04097  6.1568


```




### DISCUSSION

Our results provide support for our prediction that natal dispersal is higher in great-tailed grackle populations that are closer to the edge of the expansion range. We find that the average levels of relatedness, as well as the number of pairs of same-sex individuals that are closely related are lower in the population in California than in the population in Arizona. Grackles have been breeding since 2004 in California and since 1951 in Arizona. Our analyses suggest that the observed differences between the two populations in the levels of relatedness are unlikely to be simply due to the larger sample of individuals included in the Arizona population. While the results support our main prediction, further assessment of the hypothesis that individuals in edge populations behave differently than those nearer the core of the range is required, because our inferences rely on only a single comparison between two populations that might also differ in other aspects besides the age at which they were established. We also find that, in both populations, females are more likely to remain closer to same-sex relatives than males, suggesting that females disperse shorter distances than males. These findings, with our larger sample from this article, confirm our previous inferences for the population in Arizona [@sevchik2021dispersal], that the sex-biases in dispersal in great-tailed grackles are the opposite to that observed in most other bird species.

In the population closer to the edge of the range in California, our relatedness analyses indicate that no pair of same-sex individuals is related at the level of cousins (r≧0.125) or higher. Our inference is based on a relatively small sample of 13 females and 15 males, which is nevertheless larger than the minimum sample size set in our preregistration. In addition, all analyses suggest that the low relatedness, and in particular the absence of related same sex dyads, is unexpected given the levels of relatedness we observe among the individuals in Arizona. While the permutation analysis suggests that there might be a chance to observe no female relatives in such a sample, this approach is limited because it does not fully take into account the potential contingencies in the observed data (for example, if a mother is present with two daughters, these dyads are not independent). We therefore performed an unregistered post-hoc analysis using a social network approach that accounts for such potential interdependence using functions of the package ‘STRAND’ in R [@ross2024modelling]. We coded whether a given pair of individuals in either population was likely kin or not (r≧0.125) and determined whether the likelihood that individuals are in a kin dyad is different between the two populations. These models also indicate that the likelihood that individuals in California are closely related is substantially lower than that of individuals in Arizona (for all individuals: 89%CI estimate of difference in likelihood 0.94 to 6.81; for females 89% CI -2.36 to 4.79; for males 89% CI -0.04 to 6.16). Our results suggest that beyond the radius that we sampled, California individuals of both sexes disperse further from where they hatched than individuals in Arizona. 

Previous theoretical and empirical studies predict such increased dispersal at the edge of a population expansion. Multiple processes could contribute to the higher dispersal at the edge of the population expansion. The higher frequency of dispersers at the edge could result from simple sorting processes, whereby highly dispersive individuals are over-represented in edge populations because they are more likely to end up in these novel areas [@shine2011evolutionary, @travis2002dispersal]. Alternatively, or in addition, the conditions at the edge could shift the trade-off of the costs and benefits towards dispersing in the edge population [@simmons2004changes, @chuang2016expanding]. Such trade-offs linked to expansion have been observed in relation to dispersal of aggressive individuals in bluebirds [@duckworth2007coupling] and morphological adaptation for speed in cane toads [@clarke2019may]. Great-tailed grackle females, who show more of a change in their dispersal patterns at the edge, are likely to have changes in their trade-offs of the costs and benefits of dispersal. At the edge, females might gain increased benefits from dispersing via reduced resource competition by moving into new areas. In contrast, nearer the core, females might benefit from local knowledge and potential kin tolerance when remaining, but would face competition even if they move because all areas are already occupied. 

In the population in Arizona, we observed a small number of pairs of individuals related at the level of cousins or higher. However, while only 3% of all dyads are related at r≧0.125, 60% of females (31 out of the 52) and 56% of males (15 out of the 27) have at least one same-sex relative in the population. The kin composition we observed among great-tailed grackles in Arizona is similar to what has been reported for ravens, where 2.2% of dyads were classified as close kin and 20% of individuals had a close kin in their foraging group (the study used a different approach to estimate relatedness so the category is between our cut-offs of 0.25 and 0.125; @parker1994common). The raven study also suggested that kinship, besides parent-offspring relations, did not play a major role in structuring social interactions. Both ravens and grackles form foraging groups, where individuals are generally resighted at the same location with the same set of others. However, groups are not closed and cohesive, unlike the stable groups found in cooperatively breeding birds or several social mammals, where levels of kinship are generally higher than what we observed here and kinship plays an important role in social relationships [@pereira2023kinship].

In both populations, we find indications of a sex bias in dispersal, with females apparently dispersing shorter distances than males. Despite the absence of close relatives in California, the analyses linking relatedness to geographic distance also supports a similar bias in this population. This confirms our previous conclusion with a smaller sample in Arizona [@sevchik2021dispersal]. We find more male relatives in Arizona than in our earlier study that used a subset of these Arizona individuals [@sevchik2021dispersal]. This indicates that, while males disperse more than females, they apparently do not move much further than the distances involved in our sampling areas (2,000m). With our approach, we cannot track individual movements. Sex biases in dispersal could either arise because, on average, all males move larger distances than all females. Alternatively, differences could arise because a higher frequency of males compared to females disperse, even though, when they disperse, both males and females move similar distances[@sutherland2000scaling]. The male bias in dispersal also matches with observational reports of which individuals are first observed at the edge of the range expansion. An earlier study found that, of the first sightings of a great-tailed grackle in a new location, where the sex of the individual was reported, the pioneer individual was a male in 65% of instances [@dinsmore1993range]. The male sex bias in great-tailed grackle dispersal is the opposite of that found in most other bird species [@greenwood1980mating] where single pairs of males and females breed monogamously, and males remain where they were hatched and females disperse. This contrast in the great-tailed grackle sex bias in dispersal matches their social and mating system of polygamous breeding in larger groups [@wehtje2003range], which is more similar to that of mammals where males generally disperse further than females [@trochet2016evolution].

The male sex bias in dispersal, and the reduced sex bias in the edge population where females also appear to disperse more, might interact with the ongoing range expansion of the great-tailed grackles. In most sexually reproducing species, the distribution and movement of females determines the range limits [@miller2022two]. Particularly in species where single males mate with multiple females, as in the great-tailed grackles, we would expect that moderate levels of female-biased dispersal would increase the range expansion speed because this would lead to the sex ratio of multiple females per mating male in the new populations [@miller2011sex]. Accordingly, the adaptability of dispersal behavior in grackles, with both sexes showing more dispersal at the edge than nearer the core, might contribute to their ability to expand into new areas.

Our study on dispersal supports the role of variability in behavior for the ability of great-tailed grackles to rapidly expand their range, in line with our other findings resulting from this preregistration. Additional studies are needed to determine the robustness and potential mechanisms involved in finding different dispersal behavior in an edge population. It is not clear whether these differences reflect the particular conditions of edge populations or other ecological conditions that could also influence dispersal behavior. The differences in dispersal behavior could also be linked to differences in other behavior, rather than directly reflecting a response to whether the individuals are at the edge or nearer the core of the distribution. Our previous comparison of several behaviors indicated  higher persistence and interindividual differences in behavioral flexibility in the edge population, but no differences in average exploration, innovativeness, or behavioral flexibility [@logan2023xpoppcj]. In addition, these analyses can not unravel whether the differences in dispersal behavior reflect population level differences in the expression of individual behavior. Alternatively, individuals with high dispersal tendencies might occur across the range, but, at the edge, those dispersing the furthest might accumulate because of the lag in arrival of individuals with lower dispersal ability. We hope that our findings will stimulate additional studies into the traits that characterize individuals and populations at the edge of population range expansions.

\newpage

### ETHICS

This research is carried out in accordance with permits from the:

1)  US Fish and Wildlife Service (scientific collecting permit number MB76700A-0,1,2)
2)  US Geological Survey Bird Banding Laboratory (federal bird banding permit number 23872)
3)  Arizona Game and Fish Department (scientific collecting license number SP594338 \[2017\], SP606267 \[2018\], SP639866 \[2019\], and SP402153 \[2020\])
4)  Institutional Animal Care and Use Committee at Arizona State University (protocol number 17-1594R)
5)  California Department of Fish and Wildlife (scientific collecting permit \[specific use\] number S‐192100001‐19210‐001)
6) Institutional Animal Care and Use Committee at the University of California Santa Barbara (protocol number 958)
7) RegionalSan access permit (number AP 2021-01)


### AUTHOR CONTRIBUTIONS

**Dieter Lukas**: Hypothesis development, data analysis and interpretation, write up, revising/editing\
**Aaron Blackwell**: data collection, data interpretation, revising/editing\
**Maryam Edrisi**: data collection, revising/editing\
**Kristin Hardy**: data collection, revising/editing\
**Zara Marfori**: data collection, revising/editing\
**Kelsey McCune**: data collection, data interpretation, revising/editing\
**August Sevchik**: data collection, data interpretation, revising/editing\
**Caroline Smith**: data collection, revising/editing\
**Corina Logan**: Hypothesis development, data collection, data interpretation, revising/editing, materials/funding.


### FUNDING

This research was funded by the Department of Human Behavior, Ecology and Culture at the Max Planck Institute for Evolutionary Anthropology.

### CONFLICT OF INTEREST DISCLOSURE

We, the authors, declare that we have no financial conflicts of interest with the content of this article. CJ Logan and D Lukas are Recommenders at PCI Ecology, and CJ Logan was on the Managing Board at PCI Ecology (2018-2022).

### ACKNOWLEDGEMENTS

We thank: our PCI Ecology recommender, Esther Sebastián González, and reviewers, Caroline Nieberding, Tim Parker, and Pizza Ka Yee Chow, for their helpful feedback on the preregistration; Luisa Bergeron, Melissa Folsom, Zoe Johnson-Ulrich, Christa LeGrande, Maggie MacPherson, and Carol Rowney for sample collection and processing; Bronwyn Butcher and the Cornell Lab of Ornithology for running ddRADseq and teaching our students in these methods; Nancy Chen for connecting us with students who wanted to learn ddRADseq; Xuewen Geng and Xin Yi He for support running ddRADseq; Woodland-Davis Clean Water Agency, RegionalSan, and Conaway Ranch for hosting the research on their land; Kristine Johnson for technical advice on great-tailed grackles; Julia Cissewski and Sophie Kaube for tirelessly solving problems involving financial transactions and contracts; Richard McElreath for project support; and Ken Kosik for being a UCSB sponsor of the Cooperation Agreement with the Max Planck Institute for Evolutionary Anthropology.


### REFERENCES
