---
title: "Investigating sex differences in genetic relatedness in great-tailed grackles in Sacramento, California to infer potential sex biases in dispersal at a range edge"
format: pdf
editor: source
author: 
- 'Lukas D^1^*'
- 'Kelsey McCune'
- 'Luisa M. Bergeron'
- 'Melissa Folsom'
- "Christa LeGrande-Rolls"
- 'Zara Marfori'
- 'Carol Rowney'
- 'Maggie MacPherson'
- 'Zoe Johnson-Ulrich'
- 'Smith, Caroline'
- 'Blackwell, Aaron D'
- 'Maryam Edrisi'
- 'Xuewen Geng'
- 'Kristin Hardy'
- 'Xin Yi He'
- 'August Sevchik'
- 'Logan CJ^1,2^'
bibliography: MyLibrary.bib
csl: https://raw.githubusercontent.com/citation-style-language/styles/master/apa.csl
urlcolor: blue
---

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE) 
#Make code wrap text so it doesn't go off the page when Knitting to PDF

knitr::opts_chunk$set(echo=F, include=T, results='asis', warning=F, message=F) 
#sets global options to display code along with the results https://exeter-data-analytics.github.io/LitProg/r-markdown.html
#set echo=F for knitting to PDF (hide code), and echo=T for knitting to HTML (show code)
```

Open... ![](logoOpenAccess.png){width="5%"} access ![](logoOpenCode.png){width="5%"} [code](https://github.com/corinalogan/grackles/blob/master/Files/Preregistrations/gxpopbehaviorhabitat.Rmd) ![](logoOpenPeerReview.png){width="5%"} peer review

**Affiliations:** 1) Max Planck Institute for Evolutionary Anthropology, 2) University of California Santa Barbara \*Corresponding author: dieter_lukas\@eva.mpg.de

**This article is based on the following preregistration:** Logan CJ, McCune KB, Chen N, Lukas D. 2020. [**Implementing a rapid geographic range expansion - the role of behavior and habitat changes**](http://corinalogan.com/Preregistrations/gxpopbehaviorhabitat.html) (http://corinalogan.com/Preregistrations/gxpopbehaviorhabitat.html)

**which has been pre-study peer reviewed and received an In Principle Recommendation by:** Esther Sebastián González (2020) The role of behavior and habitat availability on species geographic expansion. *Peer Community in Ecology*, 100062. [10.24072/pci.ecology.100062](https://doi.org/10.24072/pci.ecology.100062). Reviewers: Caroline Nieberding, Tim Parker, and Pizza Ka Yee Chow

![](logoPCIecology.png){width="50%"}

### ABSTRACT

It is generally thought that behavioral flexibility, the ability to change behavior when circumstances change, plays an important role in the ability of a species to rapidly expand their geographic range [e.g. @lefebvre1997feeding; @griffin2014innovation; @chow2016practice; @sol2000behavioural; @sol2002behavioural; @sol2005big; @sol2007big]. However, it is an alternative non-exclusive possibility that an increase in the amount of available habitat can also facilitate a range expansion [@hanski1991metapopulation; @wiens1997metapopulation]. Great-tailed grackles (*Quiscalus mexicanus*) are a social, polygamous species that is rapidly expanding its geographic range [@wehtje2003range] and eats a variety of human foods in addition to foraging on insects and on the ground for other natural food items [@johnson2001great]. They are behaviorally flexible [@logan2016flexibilityproblem] and highly associated with human-modified environments [@johnson2001great], thus offering an opportunity to assess the role of behavior and habitat change over the course of their expansion. We first aim to compare behavior in wild-caught grackles from three populations across their range (core of the original range, a more recent population in the middle of the northern expansion front, a very recent population on the northern edge of the expansion front) to investigate whether: 1) certain behaviors (flexibility, innovativeness, exploration, and persistence) have higher averages and variances in some populations relative to others, and 2) individuals in a more recently established population exhibit more dispersal behavior (i.e., individuals are more likely to move away from their parents). Secondly, we aim to investigate whether habitat availability, not necessarily inherent species differences, can explain why great-tailed grackles are able to much more rapidly expand their range than their closest relative, boat-tailed grackles (*Q. major*) [@post1996boat; @wehtje2003range]. We will examine temporal habitat changes over the past few decades using existing databases on presence/absence of both grackle species and compare habitat variables to determine whether: 3) these species use different habitats, habitat suitability and connectivity (which combined determines whether habitat is available) has increased across their range, and what proportion of suitable habitat both species occupy. **Finally, we will 4) determine whether changes in behavioral traits facilitate the rapid GTGR expansion by comparing their behavior with BTGR on the same tests in aim 1.** Results will elucidate whether the rapid geographic range expansion of great-tailed grackles is associated with individuals differentially expressing particular behaviors and/or whether the expansion is facilitated by the alignment of their natural behaviors with an increase in suitable habitat (i.e., human-modified environments).

### INTRODUCTION

It is generally thought that behavioral flexibility, the ability to change behavior when circumstances change [see @mikhalevich_is_2017 for theoretical background on our flexibility definition], plays an important role in the ability of a species to rapidly expand their geographic range [e.g., @lefebvre1997feeding; @griffin2014innovation; @chow2016practice; @sol2000behavioural; @sol2002behavioural; @sol2005big; @sol2007big]. These ideas predict that flexibility, exploration, and innovation facilitate the expansion of individuals into completely new areas and that their role diminishes after a certain number of generations [@wright2010behavioral]. In support of this, experimental studies have shown that latent abilities are primarily expressed in a time of need [e.g., @taylor2007spontaneous; @bird2009insightful; @manrique2011spontaneous; @auersperg2012spontaneous; @laumer2018spontaneous]. Therefore, we do not expect the founding individuals who initially dispersed out of their original range to have unique behavioral characteristics that are passed on to their offspring. Instead, we expect that the actual act of continuing a range expansion relies on flexibility, exploration, innovation, and persistence, and that these behaviors are therefore expressed more on the edge of the expansion range where there have not been many generations to accumulate relevant knowledge about the environment.

To determine whether behavior is involved in a rapid geographic range expansion, direct measures of individual behavioral abilities must be collected in populations across the range of the species [see the discussion on the danger of proxies of flexibility in @logan2018beyond]. We plan to test whether behavioral flexibility and/or an increase in habitat availability play a role in the rapid geographic range expansion of great-tailed grackles (*Quiscalus mexicanus*). Great-tailed grackles are behaviorally flexible [@logan2016flexibilityproblem], rapidly expanding their geographic range [@wehtje2003range], and highly associated with human-modified environments [@johnson2001great], thus offering an opportunity to assess the role of behavior and habitat change over the course of their expansion. This social, polygamous species eats a variety of human foods in addition to foraging on insects and on the ground for other natural food items [@johnson2001great]. This feature increases the ecological relevance of comparative cognition experiments that measure individual behavior abilities: grackles eat at outdoor cafes, from garbage cans, and they eat our crops. As such, they generally gain experience in the wild with approaching and opening novel objects to seek food (e.g., attempting to open a ketchup packet at an outdoor cafe, climbing into garbage cans to get french fries at the zoo, dunking sugar packets in water), which makes the tests involving human-made apparatuses ecologically relevant for this species.

We here compare behavior in wild-caught great-tailed grackles from two populations across their range (a more recent population in the middle of the northern expansion front: Arizona, a very recent population on the northern edge of the expansion front: California). We investigated whether certain behaviors have higher averages and variances in the edge population relative to older populations. Specifically, in this article, we examine whether individuals in a recently established population (California) are more likely to move away from the location they hatched by determining whether their average relatedness (calculated using single nucleotide polymorphisms, SNPs) is lower than what we would expect if individuals move randomly [@sevchik2019dispersal].

There could be multiple mechanisms underpinning the results we find, however our aim is to narrow down the role of changes in behavior and changes in habitats in the range expansion of great-tailed grackles. Results will elucidate whether the rapid geographic range expansion of great-tailed grackles is associated with individuals differentially expressing particular behaviors and/or whether the expansion is facilitated by the alignment of their natural behaviors with an increase in suitable habitat (i.e., human-modified environments).

### PARTITIONING THE RESULTS

Results addressing the additional research questions from the preregistrations are presented in:

Summers et al Logan & Lukas BTGR Logan et al. McCune et al.

### RESEARCH QUESTIONS

#### Q2 (dispersal behavior: great-tailed grackles): Are there differences in dispersal behavior across the great-tailed grackle's geographic range? (Fig. 1, Table 1)

**Prediction 2:** We predict **more dispersal at the edge**: a higher proportion of individuals, particularly females, which is the sex that appears to be philopatric in the middle of the range expansion [@sevchik2019dispersal], disperse in a more recently established population and, accordingly, fewer individuals are closely related to each other. *This would support the hypothesis* that changes in dispersal behavior are involved in the great-tailed grackle's geographic range expansion.

**Prediction 2 alternative 1:** We predict that the **proportion of individuals dispersing is not related to when the population established** at a particular site and, accordingly, the average relatedness is similar across populations. *This supports the hypothesis* that the original dispersal behavior was already well adapted in this species to facilitate a range expansion.

**Table 1.** Population characteristics for each of the three field sites in Q1 and Q2. The number of generations at a site is based on a generation length of 5.6 years for this species [@GTGRbirdlife2018; note that this species starts breeding at age 1], and on the first year in which this species was reported (or estimated) to breed at each location (Woodland, California: Yolo Audubon Society's newsletter *The Burrowing Owl* (July 2004), which Steve Hampton shared with Logan; Tempe, Arizona: estimated based on 1945 first-sighting report in nearby Phoenix, Arizona [@wehtje2004thesis] to which we added 6 years, which is the average time between first-sighting and first-breeding - see Table 3 in [@wehtje2003range]; Central America: there is no data on the first year in which they started breeding because this species originates in this region, therefore we used the age of the species: 800,000 years [@johnsoncicero2004new]).

```{r eval=TRUE, warning=FALSE, results='asis', include=TRUE}
#| label: table1
#| tbl-colwidths: [60,60,30,30,30,60]
d <- read.csv(url("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/gxpopbehaviorhabitat_Table1.csv"), header=F, sep=",", stringsAsFactors=F) 

d<-d[2:3,]
colnames(d) <- c("Site","Range position","Breeding since","Number of years breeding","Average number of generations","Citation")

knitr::kable(d, row.names = FALSE)
```

### STATE OF THE DATA AND CHANGES FROM PREREGISTRATION

The preregistration was written (Mar 2020) prior to collecting any data from the edge and core populations, therefore we were blind to these data. However, we were not blind to some of the data from the Arizona population: some of the relatedness data (SNPs used for Hypothesis 2 to quantify relatedness to infer whether individuals disperse away from relatives) from the middle population (Arizona) has already been analyzed for other purposes (n=57 individuals, [see](http://corinalogan.com/Preregistrations/gdispersal_manuscript.html) @sevchik2019dispersal). Therefore, it will be considered secondary data: data that are in the process of being collected for other investigations. We have now collected blood samples from many more grackles in Arizona, therefore we will redo the analyses from the Arizona population in the analyses involved in the current preregistration. In May 2020, we completed data collection for other variables at the Arizona field site: [flexibility and innovation](http://corinalogan.com/Preregistrations/g_flexmanip.html) [@logan2019flexmanip], and [exploration](http://corinalogan.com/Preregistrations/g_exploration.html) [@mccune2019exploration], and we will soon analyze this data, therefore it will also be considered secondary data. This preregistration was submitted in May 2020 to PCI Ecology for pre-study peer review. We received the reviews, and revised and resubmitted in Aug 2020, and it passed pre-study peer review in Oct 2020.

**Level of data blindness:** Logan and McCune collect the behavioral data (Q1) and therefore have seen this data for the Arizona population. Lukas has access to the Arizona data and has seen some of the summaries in presentations. Chen has not seen any data.

-   Due to various reasons, we did not collect data for the third population from the core of the original distribution.

### METHODS

#### Sample

**Q2:** Great-tailed grackles were caught in the wild in Woodland and Sacramento, California. Adults were identified from their eye color, which changes from brown to yellow upon reaching adulthood (Johnson and Peer 2001). We applied colored leg bands in unique combinations for individual identification. Some individuals (\~20) were brought temporarily into aviaries for behavioral choice tests, and then were released back to the wild at their point of capture. We caught grackles with a variety of methods (e.g., walk-in traps, mist nets, bow nets), some of which decrease the likelihood of a selection bias for exploratory and bold individuals because grackles cannot see the traps (i.e., mist nets).

While the above is our ideal plan, due to restrictions around COVID-19, it may not be possible for us to accomplish all of our goals within our current funding period. We think it will be possible to collect data at one more site (which would be the second of three planned sites) and we will attempt to also include a third field site.

#### Sample size rationale

**Q2:** We test as many great-tailed grackles as we can during the approximately one year we spend at each site given that the birds are only brought into the aviaries during the non-breeding season (approximately September through March). It is time intensive to conduct the aviary test battery (2-6 months per bird at the Arizona field site), therefore we approximate that the minimum sample size at each site will follow the minimum sample sizes in Table 2 with the aim that half of the grackles tested at each site are female. However, we expect to be able to test 20 grackles per site.

#### Data collection stopping rule

We will stop collecting data on wild-caught great-tailed grackles in Q1 and Q2 (data for Q3 are collected from the literature) once we have completed one year at each of the California and Central America sites (likely complete in summer 2022), which coincides with the period in which we currently have funding (until early 2023). If we are not able to collect data at a third site, we will attempt to collect more data during a second year at the second site (Woodland, CA).

#### Protocols and open materials

-   **Dispersal:** DNA is collected from the grackles, processed, and analyzed for pairwise relatedness using ddRADseq and Stacks as in @sevchik2019dispersal ([protocol](http://corinalogan.com/Preregistrations/gdispersal_manuscript.html)).

#### Open data

All data for analyses are available in the Knowledge Network for Biocomplexity's data repository. Raw genotype files are available in the Sequence Read Archive of the National Center for Biotechnology Information.

#### Blinding during analysis

Blinding is usually not involved in the final analyses because the experimenters collect the data (and therefore have seen some form of it) and run the analyses.

### E. ANALYSIS PLAN

We use **simulations** and design customized **models** to determine what sample sizes allow us to detect differences between sites (see chapter 5.3 in @bolker2008ecological for why simulations perform more powerful power analyses). We do not plan to **exclude** any data and if there are **missing** data (e.g. if a bird participated in one of the two experiments, then it will only be included in those analyses for which it has data). Analyses will be conducted in R (current version `r getRversion()`; @rcoreteam) and Stan (version 2.18, @carpenter2017stan).

#### Q2: dispersal

*Response variable*

1)  Average relatedness between all pairs of individuals within one sex

*Explanatory variables*

1)  Site diameter (meters)

2)  Site sample size

3)  Number of generations at a site

One model will be run per sex

The data will be analyzed as in @sevchik2019dispersal. To summarize, blood is collected from the bird, DNA is extracted (by Aaron Blackwell at Washington State University), size selected (between 400-700 base pairs), and sequenced using ddRADseq (at Cornell University Lab of Ornithology) on an Illumina NextSeq500 machine using the mid-output setting for 150 base pair single end reads. Data are post processed to generate single nucleotide polymorphisms (SNPs) as in @thrasher2018double. Genetic relatedness between all pairs of individuals is calculated using the package “related” (@pew2015related) in R (as in @thrasher2018double) using the estimator by Queller & Goodnight, which was more robust for our inferences in a subset of the Arizona data [@sevchik2019dispersal]. Permutations (i.e., randomly assigning site ID to individuals) and general linear models estimating average relatedness of each individual to all others at that site will be used to determine whether individuals at one site are more closely related to each other than the individuals at another site.





**Model and simulation**

Expected values for average relatedness per bird were based on the fact that average relatedness with these estimators has to range between -1 and 1 and because it is an average we expect a normal distribution.

averagerelatedness \~ $\alpha$\[site\] *\[the model\]*

$\alpha$\[site\] \~ Normal($\mu$,$\sigma$) *\[*$\alpha_1$ prior\]

$\mu$ \~ Normal(0,1) *\[*$\mu$ prior\]

$\sigma$ \~ Uniform(0,1) *\[*$\sigma$ prior\]

```{r relatedness calculation, include=FALSE, echo=FALSE}
#Load the relevant packages
library(related)
library(tidyr)
library(dplyr)
library(vegan)
library(geosphere)
library(rethinking)

# Compare AZ sample size to Gus article - are the previously genotyped individuals included? Does the one indivdiual have to be excluded again?

#Load the relevant data
#SNP data, processed to be in correct format to calculate pairwise relatedness (output files saved for flexforaging article)
ca_input<-readgenotypedata("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/g_flexforaging_data_genepop_Woodland.txt")

az_input<-readgenotypedata("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/g_flexforaging_data_genepop_Tempe.txt")

# Load additional information on individuals
idinfo <- read.csv(url("https://raw.githubusercontent.com/corinalogan/grackles/master/Files/Preregistrations/gxpopbehaviorhabitatq2dispersal_individualinfoforrelatedness.csv"), header=T, sep=",", stringsAsFactors=F)

# remove an individual for which we did not have blood, and therefore no genotype
idinfo<-idinfo[-38,]

# Exclude juveniles from the dataset, because they might not yet have dispersed
adultids<-idinfo[idinfo$Age %in% "A",]

samplesize<-as.data.frame(adultids %>% group_by(Population,Sex) %>% summarize(samplesize=n()))

ca_snpdata<-ca_input$gdata
ca_snpdata$V1<-gsub("-","",ca_snpdata$V1)
ca_snpdata<-ca_snpdata[ca_snpdata$V1 %in% adultids$BirdID,]

az_snpdata<-az_input$gdata
az_snpdata$V1<-gsub("A-","A",az_snpdata$V1)
az_snpdata<-az_snpdata[az_snpdata$V1 %in% adultids$BirdID,]

# individual A053PS is included twice - because we resequenced this individual
az_snpdata<-az_snpdata[-45,] # remove original
# az_snpdata<-az_snpdata[-54,] # remove new, as alternative because the reprocessing appears to have fixed the previous problems

#Calculate pairwise relatedness, here choosing the relatedness method developed by Wang, Queller & Goodknight, Lynch & Li
ca_outfile<-coancestry(ca_snpdata,wang=1,quellergt=1,lynchli=1)
az_outfile<-coancestry(az_snpdata,wang=1,quellergt=1,lynchli=1)

#extract the relevant information from the file
ca_pairwise_r<-ca_outfile$relatedness
az_pairwise_r<-az_outfile$relatedness

# List for each individuals their pairwise relatedness to all other members of the group
ca_completerelatedness<-as.data.frame(matrix(nrow=2*nrow(ca_pairwise_r),ncol=ncol(ca_pairwise_r)))
colnames(ca_completerelatedness)<-colnames(ca_pairwise_r)
ca_completerelatedness$pair.no<-c(ca_pairwise_r$pair.no,ca_pairwise_r$pair.no)
ca_completerelatedness$ind1.id<-c(ca_pairwise_r$ind1.id,ca_pairwise_r$ind2.id)
ca_completerelatedness$ind2.id<-c(ca_pairwise_r$ind2.id,ca_pairwise_r$ind1.id)
ca_completerelatedness$group<-c(ca_pairwise_r$group,ca_pairwise_r$group)
ca_completerelatedness$trioml<-c(ca_pairwise_r$trioml,ca_pairwise_r$trioml)
ca_completerelatedness$wang<-c(ca_pairwise_r$wang,ca_pairwise_r$wang)
ca_completerelatedness$lynchli<-c(ca_pairwise_r$lynchli,ca_pairwise_r$lynchli)
ca_completerelatedness$lynchrd<-c(ca_pairwise_r$lynchrd,ca_pairwise_r$lynchrd)
ca_completerelatedness$ritland<-c(ca_pairwise_r$ritland,ca_pairwise_r$ritland)
ca_completerelatedness$quellergt<-c(ca_pairwise_r$quellergt,ca_pairwise_r$quellergt)
ca_completerelatedness$dyadml<-c(ca_pairwise_r$dyadml,ca_pairwise_r$dyadml)



az_completerelatedness<-as.data.frame(matrix(nrow=2*nrow(az_pairwise_r),ncol=ncol(az_pairwise_r)))
colnames(az_completerelatedness)<-colnames(az_pairwise_r)
az_completerelatedness$pair.no<-c(az_pairwise_r$pair.no,az_pairwise_r$pair.no)
az_completerelatedness$ind1.id<-c(az_pairwise_r$ind1.id,az_pairwise_r$ind2.id)
az_completerelatedness$ind2.id<-c(az_pairwise_r$ind2.id,az_pairwise_r$ind1.id)
az_completerelatedness$group<-c(az_pairwise_r$group,az_pairwise_r$group)
az_completerelatedness$trioml<-c(az_pairwise_r$trioml,az_pairwise_r$trioml)
az_completerelatedness$wang<-c(az_pairwise_r$wang,az_pairwise_r$wang)
az_completerelatedness$lynchli<-c(az_pairwise_r$lynchli,az_pairwise_r$lynchli)
az_completerelatedness$lynchrd<-c(az_pairwise_r$lynchrd,az_pairwise_r$lynchrd)
az_completerelatedness$ritland<-c(az_pairwise_r$ritland,az_pairwise_r$ritland)
az_completerelatedness$quellergt<-c(az_pairwise_r$quellergt,az_pairwise_r$quellergt)
az_completerelatedness$dyadml<-c(az_pairwise_r$dyadml,az_pairwise_r$dyadml)

az_individualrelatedness<-as.data.frame(az_completerelatedness %>% group_by(ind1.id) %>% summarise(average_r=mean(quellergt)))

# Check distribution of relatedness values
hist(az_completerelatedness$wang)
hist(ca_completerelatedness$wang)

```


```{r descriptive statistics, include=FALSE, echo=FALSE}
# Overall sample size
samplesize<-as.data.frame(adultids %>% group_by(Population,Sex) %>% summarize(samplesize=n()))

# missing data in the genotypes
# California
sum(ca_snpdata==0)
sum(ca_snpdata==0)/(nrow(ca_snpdata)*(ncol(ca_snpdata)-1)) # 3.3% missing data
apply(ca_snpdata[,2:986], 1, function(x) length(which(x=="0")))

# Arizona
sum(az_snpdata==0)
sum(az_snpdata==0)/(nrow(az_snpdata)*(ncol(az_snpdata)-1)) # 3.0% missing data
apply(az_snpdata[,2:925], 1, function(x) length(which(x=="0")))

# Maybe remove Buñuelo C116RY because of missing data?

```



```{r compare average relatedness, include=FALSE, echo=FALSE}
#identify which individuals are female and which are male
females<-filter(adultids,Sex %in% "F",Age %in% "A")[,]$BirdID
males<-filter(adultids,Sex %in% "M",Age %in% "A")[,]$BirdID

#Calculate average of and variance in relatedness among all individuals, all females, and all males
# First using the relatedness estimates based on the method by Wang
mean(filter(ca_completerelatedness,ind1.id %in% females,ind2.id %in% females)$wang)
mean(filter(ca_completerelatedness,ind1.id %in% males,ind2.id %in% males)$wang)
mean(ca_completerelatedness$wang)

var(filter(ca_completerelatedness,ind1.id %in% females,ind2.id %in% females)$wang)
var(filter(ca_completerelatedness,ind1.id %in% males,ind2.id %in% males)$wang)
var(ca_completerelatedness$wang)


mean(filter(az_completerelatedness,ind1.id %in% females,ind2.id %in% females)$wang)
mean(filter(az_completerelatedness,ind1.id %in% males,ind2.id %in% males)$wang)
mean(az_completerelatedness$wang)

var(filter(az_completerelatedness,ind1.id %in% females,ind2.id %in% females)$wang)
var(filter(az_completerelatedness,ind1.id %in% males,ind2.id %in% males)$wang)
var(az_completerelatedness$wang)

# Next using the relatedness estimates based on the method by Queller and Goodnight
mean(filter(ca_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
mean(filter(ca_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
mean(ca_completerelatedness$quellergt)

var(filter(ca_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
var(filter(ca_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
var(ca_completerelatedness$quellergt)


mean(filter(az_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
mean(filter(az_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
mean(az_completerelatedness$quellergt)

var(filter(az_completerelatedness,ind1.id %in% females,ind2.id %in% females)$quellergt)
var(filter(az_completerelatedness,ind1.id %in% males,ind2.id %in% males)$quellergt)
var(az_completerelatedness$quellergt)


# Compare whether average relatedness is higher in one of the populations

birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
az_completerelatedness<-left_join(az_completerelatedness,birdsex,by="ind1.id")
ca_completerelatedness<-left_join(ca_completerelatedness,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
az_completerelatedness<-left_join(az_completerelatedness,birdsex,by="ind2.id")
ca_completerelatedness<-left_join(ca_completerelatedness,birdsex,by="ind2.id")
az_completerelatedness$sexdyad<-paste(az_completerelatedness$ind1.sex,az_completerelatedness$ind2.sex)
ca_completerelatedness$sexdyad<-paste(ca_completerelatedness$ind1.sex,ca_completerelatedness$ind2.sex)
az_completerelatedness[which(az_completerelatedness$sexdyad=="F F"),]$sexdyad<-"1"
az_completerelatedness[which(az_completerelatedness$sexdyad=="M F"),]$sexdyad<-"2"
az_completerelatedness[which(az_completerelatedness$sexdyad=="F M"),]$sexdyad<-"2"
az_completerelatedness[which(az_completerelatedness$sexdyad=="M M"),]$sexdyad<-"3"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="F F"),]$sexdyad<-"1"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="M F"),]$sexdyad<-"2"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="F M"),]$sexdyad<-"2"
ca_completerelatedness[which(ca_completerelatedness$sexdyad=="M M"),]$sexdyad<-"3"


dat_relatedness<-list(
  relatedness = c(az_completerelatedness$quellergt,ca_completerelatedness$wang),
  population = as.numeric(as.factor(c(rep(1,nrow(az_completerelatedness)),rep(2,nrow(ca_completerelatedness))))),
  sex = as.numeric(as.factor(c(az_completerelatedness$sexdyad,ca_completerelatedness$sexdyad)))
)

set.seed(1109)
mimmodel <- ulam( alist(
  relatedness ~dnorm(nu,theta),
  nu <- p[population]+s[sex,population],
  p[population]~dnorm(0,1),
  vector[2]:s[sex] ~ multi_normal(0,Rho_s,sigma_s),
  Rho_s~dlkjcorr(4),
  sigma_s~dexp(1),
  a[sex]~dnorm(0,0.1), #relatedness values are centered on zero
  theta ~ dexp(1)
) , data=dat_relatedness, chains=4 , cores=4 , cmdstan = TRUE)

precis(mimmodel,depth=2)
#             mean   sd  5.5% 94.5% rhat ess_bulk
# p[1]       -0.01 0.01 -0.02  0.00 1.01   463.20
# p[2]       -0.01 0.01 -0.02  0.00 1.02   256.30
# sigma_s[1]  0.01 0.02  0.00  0.03 1.01   333.59
# sigma_s[2]  0.01 0.04  0.00  0.04 1.01   207.15
# a[1]        0.00 0.10 -0.15  0.16 1.00  1627.24
# a[2]        0.00 0.10 -0.17  0.16 1.00  1506.39
# a[3]        0.00 0.10 -0.15  0.16 1.00  1396.31
# theta       0.08 0.00  0.08  0.08 1.00  2051.99


```


```{r relatedness permutation, include=FALSE, echo=FALSE}
#Perform a simulation to assess whether average relatedness among males is different from what we would expect in a random subset of the same number of individuals

# Arizona
#First based on the relatedness estimates based on the method by Wang
simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Tempe",]$BirdID,samplesize[4,3])
  simulatedrelatedness[i,1]<-mean(filter(az_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$wang)
}
hist(simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_malerelatednesswang_AZ<-sum(simulatedrelatedness>mean(filter(az_pairwise_r,ind1.id %in% males,ind2.id %in% males)$wang))/10000


#Perform a simulation to assess whether average relatedness among females is different from what we would expect in a random subset of the same number of individuals
simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Tempe",]$BirdID,samplesize[3,3])
  simulatedrelatedness[i,1]<-mean(filter(az_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$wang)
}
hist(simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_femalerelatednesswang_AZ<-sum(simulatedrelatedness>mean(filter(az_pairwise_r,ind1.id %in% females,ind2.id %in% females)$wang))/10000


#Next based on the relatedness estimates based on the method by Queller & Goodnight
#Perform a simulation to assess whether average relatedness among males is different from what we would expect in a random subset of the same number of individuals
simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Tempe",]$BirdID,samplesize[4,3])
  simulatedrelatedness[i,1]<-mean(filter(az_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$quellergt)
}
hist(simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_malerelatednessquellergt_AZ<-sum(simulatedrelatedness>mean(filter(az_pairwise_r,ind1.id %in% males,ind2.id %in% males)$quellergt))/10000


#Perform a simulation to assess whether average relatedness among females is different from what we would expect in a random subset of the same number of individuals
simulatedrelatedness<-matrix(ncol=1,nrow=10000)
for (i in 1:10000) {
  currentset<-sample(adultids[adultids$Population %in% "Tempe",]$BirdID,samplesize[3,3])
  simulatedrelatedness[i,1]<-mean(filter(az_pairwise_r,ind1.id %in% currentset,ind2.id %in% currentset)$quellergt)
}
hist(simulatedrelatedness)
#This value is similar to a p-value, it reflects the probability that the average relatedness observed among males would be expected in a random subsample
p_simulation_femalerelatednessquellergt_AZ<-sum(simulatedrelatedness>mean(filter(az_pairwise_r,ind1.id %in% females,ind2.id %in% females)$quellergt))/10000


az_closerelatives<-az_pairwise_r[az_pairwise_r$quellergt>0.249,]
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
az_closerelatives<-left_join(az_closerelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
az_closerelatives<-left_join(az_closerelatives,birdsex,by="ind2.id")
nrow(az_closerelatives[az_closerelatives$ind1.sex=="F" & az_closerelatives$ind2.sex=="F",])
nrow(az_closerelatives[az_closerelatives$ind1.sex=="F" & az_closerelatives$ind2.sex=="M",])
nrow(az_closerelatives)

# 10 close female-female relatives, 6 close male-male relatives, 9 male-female relatives

az_closerelatives<-az_pairwise_r[az_pairwise_r$wang>0.249,]
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
az_closerelatives<-left_join(az_closerelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
az_closerelatives<-left_join(az_closerelatives,birdsex,by="ind2.id")
nrow(az_closerelatives[az_closerelatives$ind1.sex=="F" & az_closerelatives$ind2.sex=="F",])
nrow(az_closerelatives[az_closerelatives$ind1.sex=="F" & az_closerelatives$ind2.sex=="M",])
nrow(az_closerelatives)

# 27 close female-female relatives, 16 close male-male relatives, 21 male-female relatives



ca_closerelatives<-ca_pairwise_r[ca_pairwise_r$quellergt>0.249,]
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
ca_closerelatives<-left_join(ca_closerelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
ca_closerelatives<-left_join(ca_closerelatives,birdsex,by="ind2.id")
nrow(ca_closerelatives[ca_closerelatives$ind1.sex=="F" & ca_closerelatives$ind2.sex=="F",])
nrow(ca_closerelatives[ca_closerelatives$ind1.sex=="F" & ca_closerelatives$ind2.sex=="M",])
nrow(ca_closerelatives)

# 0 close female-female relatives, 0 close male-male relatives, 2 male-female relatives

ca_closerelatives<-ca_pairwise_r[ca_pairwise_r$wang>0.249,]
birdsex<-adultids[,c(2,5)]
colnames(birdsex)<-c("ind1.id","ind1.sex")
ca_closerelatives<-left_join(ca_closerelatives,birdsex,by="ind1.id")
colnames(birdsex)<-c("ind2.id","ind2.sex")
ca_closerelatives<-left_join(ca_closerelatives,birdsex,by="ind2.id")
nrow(ca_closerelatives[ca_closerelatives$ind1.sex=="F" & ca_closerelatives$ind2.sex=="F",])
nrow(ca_closerelatives[ca_closerelatives$ind1.sex=="F" & ca_closerelatives$ind2.sex=="M",])
nrow(ca_closerelatives)

# 0 close female-female relatives, 1 close male-male relatives, 6 male-female relatives

# Do the cross-population permutation - if we draw X individuals (females in California) from the sample of females in Arizona, how often do we get no close relative?


females_az_relatedness<-filter(az_completerelatedness,ind1.id %in% females,ind2.id %in% females)

counts<-NA
for(i in 1:10000){
subsetrel<-sample(females_az_relatedness$quellergt,samplesize[5,3])
counts<-c(counts,sum(subsetrel>0.249))
}
counts<-counts[2:10001]
sum(counts==0)/10000


males_az_relatedness<-filter(az_completerelatedness,ind1.id %in% males,ind2.id %in% males)

counts<-NA
for(i in 1:10000){
subsetrel<-sample(males_az_relatedness$quellergt,samplesize[6,3])
counts<-c(counts,sum(subsetrel>0.249))
}
counts<-counts[2:10001]
sum(counts==0)/10000





# Network model, determining whether the number of close relatives differs by sex and population

library(STRAND)


az_relatednessmatrix<-as.data.frame(pivot_wider(az_completerelatedness,id_cols=ind1.id,names_from=ind2.id,values_from=wang))

row.names(az_relatednessmatrix)<-az_relatednessmatrix$ind1.id
az_relatednessmatrix<-az_relatednessmatrix[,2:80]
az_relatednessmatrix<-az_relatednessmatrix[,c(79,1:78)]

az_birds<-adultids[adultids$Population %in% "Tempe",]
az_birds<-az_birds[order(az_birds$BirdID),]


az_relatives<-(az_relatednessmatrix>0.249)*1
diag(az_relatives)<-0
outcome=list(Relative=az_relatives)
group_ids=data.frame(Sex=as.factor(az_birds$Sex))


dat = make_strand_data(
  outcome = outcome,
  block_covariates = group_ids ,
  individual_covariates = NULL,
  dyadic_covariates = NULL,
  outcome_mode = "poisson"
  )


fit =
  fit_block_plus_social_relations_model(
    data=dat,
    block_regression = ~ Sex,
    focal_regression = ~ 1,
    dyad_regression = ~ 1,
    target_regression = ~ 1,
    mode="mcmc",
    stan_mcmc_parameters = list(
      chains = 1,
      iter_warmup = 1500,
      iter_sampling = 1500)
    )



# Summarize results
res = summarize_strand_results(fit)
# Plot slopes
vis = strand_caterpillar_plot(res,normalized=TRUE, only_slopes=TRUE)
vis

# Individuals of the same sex are slightly more likely to be related than individuals of the opposite sex, f-f dyads are equally likely to be close kin than are m-m dyads.

#                                             Variable Median HPDI:0.05 HPDI:0.95   Mean    SD
# 1                                   focal effects sd  1.636     1.196     2.076  1.654 0.277
# 2                                  target effects sd  1.631     1.239     2.144  1.654 0.278
# 3                                  dyadic effects sd   2.65     2.188     3.172  2.673 0.307
# 4  focal-target effects rho (generalized recipocity)  0.976     0.941     0.998  0.969 0.024
# 5             dyadic effects rho (dyadic recipocity)  0.986     0.969     0.999  0.983 0.012
# 6                                 offset, Any to Any -4.281    -6.269    -1.918 -4.262  1.35
# 7                                     offset, F to F -5.157    -7.263    -3.222 -5.184 1.224
# 8                                     offset, F to M -5.182    -7.126    -3.222 -5.219 1.193
# 9                                     offset, M to F -5.215    -7.146    -3.278 -5.227  1.19
# 10                                    offset, M to M -5.391    -7.729    -3.262 -5.389  1.35


```




In this text, we show the summary statistics: Females in Arizona `r samplesize[3,3]`, Males in Arizona `r samplesize[4,3]`, Females in California `r samplesize[5,3]`, Males in California `r samplesize[6,3]`




```{r distance calculation, include=FALSE, echo=FALSE}
# Distance between trap sites (meters): straight line distance (assuming earth as an ellipsoid) between all pairs of trapping locations based on the longitude and latitude of each site.

#Plot pairwise distances among all females and among all males in the sample
#Calculate all pairwise distances
all_pairwise_distances <- distm(adultids[,c('Lon','Lat')], adultids[,c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(all_pairwise_distances)<-adultids$Name
colnames(all_pairwise_distances)<-adultids$Name
diag(all_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult females in California
CA_female_pairwise_distances <- distm(adultids[adultids$Sex=="F" & adultids$Population=="Woodland",c('Lon','Lat')], adultids[adultids$Sex=="F" & adultids$Population=="Woodland",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(CA_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Woodland",]$BirdID
colnames(CA_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Woodland",]$BirdID
diag(CA_female_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult females in Arizona
AZ_female_pairwise_distances <- distm(adultids[adultids$Sex=="F" & adultids$Population=="Tempe",c('Lon','Lat')], adultids[adultids$Sex=="F" & adultids$Population=="Tempe",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(AZ_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Tempe",]$BirdID
colnames(AZ_female_pairwise_distances)<-adultids[adultids$Sex=="F" & adultids$Population=="Tempe",]$BirdID
diag(AZ_female_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult males in California
CA_male_pairwise_distances <- distm(adultids[adultids$Sex=="M" & adultids$Population=="Woodland",c('Lon','Lat')], adultids[adultids$Sex=="M" & adultids$Population=="Woodland",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(CA_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Woodland",]$BirdID
colnames(CA_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Woodland",]$BirdID
diag(CA_male_pairwise_distances)<-NA

#Calculate pairwise distances among all the adult males in Arizona
AZ_male_pairwise_distances <- distm(adultids[adultids$Sex=="M" & adultids$Population=="Tempe",c('Lon','Lat')], adultids[adultids$Sex=="M" & adultids$Population=="Tempe",c('Lon','Lat')], fun=distVincentyEllipsoid)
rownames(AZ_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Tempe",]$BirdID
colnames(AZ_male_pairwise_distances)<-adultids[adultids$Sex=="M" & adultids$Population=="Tempe",]$BirdID
diag(AZ_male_pairwise_distances)<-NA

#plot distributions of pairwise distances
hist(all_pairwise_distances,col="grey75",border="black",breaks=10)
hist(CA_female_pairwise_distances,col="grey75",border="black",breaks=10)
hist(CA_male_pairwise_distances,col="grey75",border="black",breaks=20)
hist(AZ_female_pairwise_distances,col="grey75",border="black",breaks=10)
hist(AZ_male_pairwise_distances,col="grey75",border="black",breaks=20)

# In California, two individuals were trapped at locations in Woodland, while the remaining birds were trapped at locations in Sacramento. Therefore, the maximum distance between trap location of individuals is much higher in California than it is in Arizona, where all birds were captured in locations around Tempe. For the analyses including pairwise distances, we will run the analyses with all California birds, and with only the birds trapped in Sacramento. 

AZ_all_pairwise_distance<-all_pairwise_distances[rownames(all_pairwise_distances) %in% adultids[adultids$Population == "Tempe",]$Name,colnames(all_pairwise_distances) %in% adultids[adultids$Population == "Tempe",]$Name]

CA_all_pairwise_distance<-all_pairwise_distances[rownames(all_pairwise_distances) %in% adultids[adultids$Population == "Woodland",]$Name,colnames(all_pairwise_distances) %in% adultids[adultids$Population == "Woodland",]$Name]



```





```{r dispersal_anaysisii, eval=FALSE, warning=FALSE, results='asis', echo=TRUE}
options(width=60)

#Analysis 2: Assess whether distances among closely related females are shorter than distances among closely related males


# Arizona
#First define close relatives as all pairs of individuals who are related by a level of 0.25 or higher (half-siblings or higher) using the Wang estimator
az_close_relatives_females<-filter(az_pairwise_r,wang >0.2499,ind1.id %in% females,ind2.id %in% females)
az_close_relatives_females_individuals<-c(az_close_relatives_females$ind1.id,az_close_relatives_females$ind2.id)

#Alternatively, select close relatives as pairs of individuals who are related at a level of 0.25 of higher using the Queller & Goodnight estimator
# az_close_relatives_females<-filter(az_pairwise_r,quellergt >0.2499,ind1.id %in% females,ind2.id %in% females)
# az_close_relatives_females_individuals<-c(az_close_relatives_females$ind1.id,az_close_relatives_females$ind2.id)

#Pick one of the two estimators before proceeding with the following analyses

#Next subset the the distance matrix to only include these individuals

az_close_relatives_females_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_females),ncol=1)

for (i in 1:nrow(az_close_relatives_females)) {
  ind1<-az_close_relatives_females[i,]$ind1.id
  ind2<-az_close_relatives_females[i,]$ind2.id
  pair_distance<-AZ_female_pairwise_distances[ind1,ind2]
  az_close_relatives_females_pairwise_distances[i,]<-pair_distance
}

median_az_f_025_distances<-median(az_close_relatives_females_pairwise_distances)

hist(az_close_relatives_females_pairwise_distances)


#repeat the same for the males
az_close_relatives_males<-filter(az_pairwise_r,wang >0.2499,ind1.id %in% males,ind2.id %in% males)
az_close_relatives_males_individuals<-c(az_close_relatives_males$ind1.id,az_close_relatives_males$ind2.id)

#Again, the alternative with the Queller & Goodnight method, pick only one of the two
# close_relatives_males<-filter(pairwise_r,quellergt >0.2499,ind1.id %in% males,ind2.id %in% males)
# close_relatives_males_individuals<-c(close_relatives_males$ind1.id,close_relatives_males$ind2.id)


#Next subset the the distance matrix to only include these individuals

az_close_relatives_males_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_males),ncol=1)

for (i in 1:nrow(az_close_relatives_males)) {
  ind1<-az_close_relatives_males[i,]$ind1.id
  ind2<-az_close_relatives_males[i,]$ind2.id
  pair_distance<-AZ_male_pairwise_distances[ind1,ind2]
  az_close_relatives_males_pairwise_distances[i,]<-pair_distance
}

median_az_m_025_distances<-median(az_close_relatives_males_pairwise_distances)

hist(az_close_relatives_males_pairwise_distances)

#calculate difference between the distances among males and among females
observeddifferenceindistances<-median(az_close_relatives_males_pairwise_distances,na.rm=T)-median(az_close_relatives_females_pairwise_distances,na.rm=T)

median_az_difference_025_distances<-observeddifferenceindistances

#perform simulation to generate random draws of matching numbers of individuals to assess whether the sex-difference in the distance is more or less than what would be expected by chance
number_close_relatives_females<-nrow(az_close_relatives_females)
number_close_relatives_males<-nrow(az_close_relatives_males)

simulateddifferencesindistances<-matrix(ncol=1,nrow=10000)
simulateddfemaleindistances<-matrix(ncol=1,nrow=10000)
simulateddmaleindistances<-matrix(ncol=1,nrow=10000)

rownames(AZ_all_pairwise_distance)<-adultids[adultids$Name %in% rownames(AZ_all_pairwise_distance),]$BirdID
colnames(AZ_all_pairwise_distance)<-adultids[adultids$Name %in% colnames(AZ_all_pairwise_distance),]$BirdID

for (i in 1:10000) {
simulated_close_relatives_females<-sample_n(az_pairwise_r, number_close_relatives_females, replace = TRUE)

subset_relatives_females_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_females),ncol=1)

for (j in 1:nrow(simulated_close_relatives_females)) {
  ind1<-simulated_close_relatives_females[j,]$ind1.id
  ind2<-simulated_close_relatives_females[j,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_females_pairwise_distances[j,]<-pair_distance
}

simulated_close_relatives_males<-sample_n(az_pairwise_r, number_close_relatives_males, replace = TRUE)

subset_relatives_males_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_males),ncol=1)

for (k in 1:nrow(simulated_close_relatives_males)) {
  ind1<-simulated_close_relatives_males[k,]$ind1.id
  ind2<-simulated_close_relatives_males[k,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_males_pairwise_distances[k,]<-pair_distance
}

simulateddfemaleindistances[i,1]<-median(subset_relatives_females_pairwise_distances,na.rm=T)
simulateddmaleindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)
simulateddifferencesindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)-median(subset_relatives_females_pairwise_distances,na.rm=T)
  }

simulation_az_f_025_distances<-sum(simulateddfemaleindistances<median(az_close_relatives_females_pairwise_distances))/10000
simulation_az_m_025_distances<-sum(simulateddmaleindistances>median(az_close_relatives_males_pairwise_distances))/10000
simulation_az_difference_025_distances<-sum(simulateddifferencesindistances>observeddifferenceindistances)/10000




# Arizona
#To match the California sample (see below), we also define close relatives as all pairs of individuals who are related by a level of 0.125 or higher (cousins or higher) using the Wang estimator
az_close_relatives_females<-filter(az_pairwise_r,wang >0.12499,ind1.id %in% females,ind2.id %in% females)
az_close_relatives_females_individuals<-c(az_close_relatives_females$ind1.id,az_close_relatives_females$ind2.id)

#Alternatively, select close relatives as pairs of individuals who are related at a level of 0.25 of higher using the Queller & Goodnight estimator
# az_close_relatives_females<-filter(az_pairwise_r,quellergt >0.2499,ind1.id %in% females,ind2.id %in% females)
# az_close_relatives_females_individuals<-c(az_close_relatives_females$ind1.id,az_close_relatives_females$ind2.id)

#Pick one of the two estimators before proceeding with the following analyses

#Next subset the the distance matrix to only include these individuals

az_close_relatives_females_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_females),ncol=1)

for (i in 1:nrow(az_close_relatives_females)) {
  ind1<-az_close_relatives_females[i,]$ind1.id
  ind2<-az_close_relatives_females[i,]$ind2.id
  pair_distance<-AZ_female_pairwise_distances[ind1,ind2]
  az_close_relatives_females_pairwise_distances[i,]<-pair_distance
}

median_az_f_0125_distances<-median(az_close_relatives_females_pairwise_distances)

hist(az_close_relatives_females_pairwise_distances)


#repeat the same for the males
az_close_relatives_males<-filter(az_pairwise_r,wang >0.12499,ind1.id %in% males,ind2.id %in% males)
az_close_relatives_males_individuals<-c(az_close_relatives_males$ind1.id,az_close_relatives_males$ind2.id)

#Again, the alternative with the Queller & Goodnight method, pick only one of the two
# close_relatives_males<-filter(pairwise_r,quellergt >0.2499,ind1.id %in% males,ind2.id %in% males)
# close_relatives_males_individuals<-c(close_relatives_males$ind1.id,close_relatives_males$ind2.id)


#Next subset the the distance matrix to only include these individuals

az_close_relatives_males_pairwise_distances<-matrix(nrow=nrow(az_close_relatives_males),ncol=1)

for (i in 1:nrow(az_close_relatives_males)) {
  ind1<-az_close_relatives_males[i,]$ind1.id
  ind2<-az_close_relatives_males[i,]$ind2.id
  pair_distance<-AZ_male_pairwise_distances[ind1,ind2]
  az_close_relatives_males_pairwise_distances[i,]<-pair_distance
}

median_az_m_0125_distances<-median(az_close_relatives_males_pairwise_distances)

hist(az_close_relatives_males_pairwise_distances)

#calculate difference between the distances among males and among females
observeddifferenceindistances<-median(az_close_relatives_males_pairwise_distances,na.rm=T)-median(az_close_relatives_females_pairwise_distances,na.rm=T)

median_az_difference_0125_distances<-observeddifferenceindistances

#perform simulation to generate random draws of matching numbers of individuals to assess whether the sex-difference in the distance is more or less than what would be expected by chance
number_close_relatives_females<-nrow(az_close_relatives_females)
number_close_relatives_males<-nrow(az_close_relatives_males)

simulateddifferencesindistances<-matrix(ncol=1,nrow=10000)
simulateddfemaleindistances<-matrix(ncol=1,nrow=10000)
simulateddmaleindistances<-matrix(ncol=1,nrow=10000)

rownames(AZ_all_pairwise_distance)<-adultids[adultids$Name %in% rownames(AZ_all_pairwise_distance),]$BirdID
colnames(AZ_all_pairwise_distance)<-adultids[adultids$Name %in% colnames(AZ_all_pairwise_distance),]$BirdID

for (i in 1:10000) {
simulated_close_relatives_females<-sample_n(az_pairwise_r, number_close_relatives_females, replace = TRUE)

subset_relatives_females_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_females),ncol=1)

for (j in 1:nrow(simulated_close_relatives_females)) {
  ind1<-simulated_close_relatives_females[j,]$ind1.id
  ind2<-simulated_close_relatives_females[j,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_females_pairwise_distances[j,]<-pair_distance
}

simulated_close_relatives_males<-sample_n(az_pairwise_r, number_close_relatives_males, replace = TRUE)

subset_relatives_males_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_males),ncol=1)

for (k in 1:nrow(simulated_close_relatives_males)) {
  ind1<-simulated_close_relatives_males[k,]$ind1.id
  ind2<-simulated_close_relatives_males[k,]$ind2.id
  pair_distance<-AZ_all_pairwise_distance[ind1,ind2]
  subset_relatives_males_pairwise_distances[k,]<-pair_distance
}

simulateddfemaleindistances[i,1]<-median(subset_relatives_females_pairwise_distances,na.rm=T)
simulateddmaleindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)
simulateddifferencesindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)-median(subset_relatives_females_pairwise_distances,na.rm=T)
  }

simulation_az_f_0125_distances<-sum(simulateddfemaleindistances<median(az_close_relatives_females_pairwise_distances))/10000
simulation_az_m_0125_distances<-sum(simulateddmaleindistances>median(az_close_relatives_males_pairwise_distances))/10000
simulation_az_difference_0125_distances<-sum(simulateddifferencesindistances>observeddifferenceindistances)/10000




# California
# In our California sample, there is only one same-sex pair of individuals who is related  at the level of 0.25 or higher. We therefore switch the calculation here to include as relatives pairs of individuals who are related at the level of 0.125 or higher (8 f-f pairs, 7 m-m pairs)
#First define close relatives as all pairs of individuals who are related by a level of 0.125 or higher (cousins or higher) using the Wang estimator
ca_close_relatives_females<-filter(ca_pairwise_r,wang >0.12499,ind1.id %in% females,ind2.id %in% females)
ca_close_relatives_females_individuals<-c(ca_close_relatives_females$ind1.id,ca_close_relatives_females$ind2.id)

#Alternatively, select close relatives as pairs of individuals who are related at a level of 0.25 of higher using the Queller & Goodnight estimator
# ca_close_relatives_females<-filter(ca_pairwise_r,quellergt >0.2499,ind1.id %in% females,ind2.id %in% females)
# ca_close_relatives_females_individuals<-c(ca_close_relatives_females$ind1.id,ca_close_relatives_females$ind2.id)

#Pick one of the two estimators before proceeding with the following analyses

#Next subset the the distance matrix to only include these individuals

ca_close_relatives_females_pairwise_distances<-matrix(nrow=nrow(ca_close_relatives_females),ncol=1)

for (i in 1:nrow(ca_close_relatives_females)) {
  ind1<-ca_close_relatives_females[i,]$ind1.id
  ind2<-ca_close_relatives_females[i,]$ind2.id
  pair_distance<-CA_female_pairwise_distances[ind1,ind2]
  ca_close_relatives_females_pairwise_distances[i,]<-pair_distance
}

median_ca_f_0125_distances<-median(ca_close_relatives_females_pairwise_distances)

hist(ca_close_relatives_females_pairwise_distances)


#repeat the same for the males
ca_close_relatives_males<-filter(ca_pairwise_r,wang >0.2499,ind1.id %in% males,ind2.id %in% males)
ca_close_relatives_males_individuals<-c(ca_close_relatives_males$ind1.id,ca_close_relatives_males$ind2.id)

#Again, the alternative with the Queller & Goodnight method, pick only one of the two
# close_relatives_males<-filter(pairwise_r,quellergt >0.2499,ind1.id %in% males,ind2.id %in% males)
# close_relatives_males_individuals<-c(close_relatives_males$ind1.id,close_relatives_males$ind2.id)


#Next subset the the distance matrix to only include these individuals

ca_close_relatives_males_pairwise_distances<-matrix(nrow=nrow(ca_close_relatives_males),ncol=1)

for (i in 1:nrow(ca_close_relatives_males)) {
  ind1<-ca_close_relatives_males[i,]$ind1.id
  ind2<-ca_close_relatives_males[i,]$ind2.id
  pair_distance<-CA_male_pairwise_distances[ind1,ind2]
  ca_close_relatives_males_pairwise_distances[i,]<-pair_distance
}

median_ca_m_0125_distances<-median(ca_close_relatives_males_pairwise_distances)

hist(ca_close_relatives_males_pairwise_distances)

#calculate difference between the distances among males and among females
observeddifferenceindistances<-median(ca_close_relatives_males_pairwise_distances,na.rm=T)-median(ca_close_relatives_females_pairwise_distances,na.rm=T)

median_ca_difference_0125_distances<-observeddifferenceindistances

#perform simulation to generate random draws of matching numbers of individuals to assess whether the sex-difference in the distance is more or less than what would be expected by chance
number_close_relatives_females<-nrow(ca_close_relatives_females)
number_close_relatives_males<-nrow(ca_close_relatives_males)

simulateddifferencesindistances<-matrix(ncol=1,nrow=10000)
simulateddfemaleindistances<-matrix(ncol=1,nrow=10000)
simulateddmaleindistances<-matrix(ncol=1,nrow=10000)

rownames(CA_all_pairwise_distance)<-adultids[adultids$Name %in% rownames(CA_all_pairwise_distance),]$BirdID
colnames(CA_all_pairwise_distance)<-adultids[adultids$Name %in% colnames(CA_all_pairwise_distance),]$BirdID

for (i in 1:10000) {
simulated_close_relatives_females<-sample_n(ca_pairwise_r, number_close_relatives_females, replace = TRUE)

subset_relatives_females_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_females),ncol=1)

for (j in 1:nrow(simulated_close_relatives_females)) {
  ind1<-simulated_close_relatives_females[j,]$ind1.id
  ind2<-simulated_close_relatives_females[j,]$ind2.id
  pair_distance<-CA_all_pairwise_distance[ind1,ind2]
  subset_relatives_females_pairwise_distances[j,]<-pair_distance
}

simulated_close_relatives_males<-sample_n(ca_pairwise_r, number_close_relatives_males, replace = TRUE)

subset_relatives_males_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_males),ncol=1)

for (k in 1:nrow(simulated_close_relatives_males)) {
  ind1<-simulated_close_relatives_males[k,]$ind1.id
  ind2<-simulated_close_relatives_males[k,]$ind2.id
  pair_distance<-CA_all_pairwise_distance[ind1,ind2]
  subset_relatives_males_pairwise_distances[k,]<-pair_distance
}

simulateddfemaleindistances[i,1]<-median(subset_relatives_females_pairwise_distances,na.rm=T)
simulateddmaleindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)
simulateddifferencesindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)-median(subset_relatives_females_pairwise_distances,na.rm=T)
  }

simulation_ca_f_0125_distances<-sum(simulateddfemaleindistances<median(ca_close_relatives_females_pairwise_distances))/10000
simulation_ca_m_0125_distances<-sum(simulateddmaleindistances>median(ca_close_relatives_males_pairwise_distances))/10000
simulation_ca_difference_0125_distances<-sum(simulateddifferencesindistances>observeddifferenceindistances)/10000



# Repeat without the birds trapped in Woodland

# The sample of birds from California includes individuals trapped at two different locations. To avoid potential sampling biases because of this, we re-run the analysis to only include individuals from the site in Sacramento, where most of the birds were trapped.



# California
# In our California sample, there is only one same-sex pair of individuals who is related  at the level of 0.25 or higher. We therefore switch the calculation here to include as relatives pairs of individuals who are related at the level of 0.125 or higher (8 f-f pairs, 7 m-m pairs)
#First define close relatives as all pairs of individuals who are related by a level of 0.125 or higher (cousins or higher) using the Wang estimator
ca_close_relatives_females<-filter(ca_pairwise_r,wang >0.12499,ind1.id %in% females,ind2.id %in% females)
ca_close_relatives_females_individuals<-c(ca_close_relatives_females$ind1.id,ca_close_relatives_females$ind2.id)

#Alternatively, select close relatives as pairs of individuals who are related at a level of 0.25 of higher using the Queller & Goodnight estimator
# ca_close_relatives_females<-filter(ca_pairwise_r,quellergt >0.2499,ind1.id %in% females,ind2.id %in% females)
# ca_close_relatives_females_individuals<-c(ca_close_relatives_females$ind1.id,ca_close_relatives_females$ind2.id)

#Pick one of the two estimators before proceeding with the following analyses

#Next subset the the distance matrix to only include these individuals

ca_close_relatives_females_pairwise_distances<-matrix(nrow=nrow(ca_close_relatives_females),ncol=1)

for (i in 1:nrow(ca_close_relatives_females)) {
  ind1<-ca_close_relatives_females[i,]$ind1.id
  ind2<-ca_close_relatives_females[i,]$ind2.id
  pair_distance<-CA_female_pairwise_distances[ind1,ind2]
  ca_close_relatives_females_pairwise_distances[i,]<-pair_distance
}

median_ca_f_0125_distances<-median(ca_close_relatives_females_pairwise_distances)

hist(ca_close_relatives_females_pairwise_distances)


#repeat the same for the males
ca_close_relatives_males<-filter(ca_pairwise_r,wang >0.2499,ind1.id %in% males,ind2.id %in% males)
ca_close_relatives_males_individuals<-c(ca_close_relatives_males$ind1.id,ca_close_relatives_males$ind2.id)

#Again, the alternative with the Queller & Goodnight method, pick only one of the two
# close_relatives_males<-filter(pairwise_r,quellergt >0.2499,ind1.id %in% males,ind2.id %in% males)
# close_relatives_males_individuals<-c(close_relatives_males$ind1.id,close_relatives_males$ind2.id)


#Next subset the the distance matrix to only include these individuals

ca_close_relatives_males_pairwise_distances<-matrix(nrow=nrow(ca_close_relatives_males),ncol=1)

for (i in 1:nrow(ca_close_relatives_males)) {
  ind1<-ca_close_relatives_males[i,]$ind1.id
  ind2<-ca_close_relatives_males[i,]$ind2.id
  pair_distance<-CA_male_pairwise_distances[ind1,ind2]
  ca_close_relatives_males_pairwise_distances[i,]<-pair_distance
}

median_ca_m_0125_distances<-median(ca_close_relatives_males_pairwise_distances)

hist(ca_close_relatives_males_pairwise_distances)

#calculate difference between the distances among males and among females
observeddifferenceindistances<-median(ca_close_relatives_males_pairwise_distances,na.rm=T)-median(ca_close_relatives_females_pairwise_distances,na.rm=T)

median_ca_difference_0125_distances<-observeddifferenceindistances

#perform simulation to generate random draws of matching numbers of individuals to assess whether the sex-difference in the distance is more or less than what would be expected by chance
number_close_relatives_females<-nrow(ca_close_relatives_females)
number_close_relatives_males<-nrow(ca_close_relatives_males)

simulateddifferencesindistances<-matrix(ncol=1,nrow=10000)
simulateddfemaleindistances<-matrix(ncol=1,nrow=10000)
simulateddmaleindistances<-matrix(ncol=1,nrow=10000)

rownames(CA_all_pairwise_distance)<-adultids[adultids$Name %in% rownames(CA_all_pairwise_distance),]$BirdID
colnames(CA_all_pairwise_distance)<-adultids[adultids$Name %in% colnames(CA_all_pairwise_distance),]$BirdID

for (i in 1:10000) {
simulated_close_relatives_females<-sample_n(ca_pairwise_r, number_close_relatives_females, replace = TRUE)

subset_relatives_females_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_females),ncol=1)

for (j in 1:nrow(simulated_close_relatives_females)) {
  ind1<-simulated_close_relatives_females[j,]$ind1.id
  ind2<-simulated_close_relatives_females[j,]$ind2.id
  pair_distance<-CA_all_pairwise_distance[ind1,ind2]
  subset_relatives_females_pairwise_distances[j,]<-pair_distance
}

simulated_close_relatives_males<-sample_n(ca_pairwise_r, number_close_relatives_males, replace = TRUE)

subset_relatives_males_pairwise_distances<-matrix(nrow=nrow(simulated_close_relatives_males),ncol=1)

for (k in 1:nrow(simulated_close_relatives_males)) {
  ind1<-simulated_close_relatives_males[k,]$ind1.id
  ind2<-simulated_close_relatives_males[k,]$ind2.id
  pair_distance<-CA_all_pairwise_distance[ind1,ind2]
  subset_relatives_males_pairwise_distances[k,]<-pair_distance
}

simulateddfemaleindistances[i,1]<-median(subset_relatives_females_pairwise_distances,na.rm=T)
simulateddmaleindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)
simulateddifferencesindistances[i,1]<-median(subset_relatives_males_pairwise_distances,na.rm=T)-median(subset_relatives_females_pairwise_distances,na.rm=T)
  }

simulation_ca_f_0125_distances<-sum(simulateddfemaleindistances<median(ca_close_relatives_females_pairwise_distances))/10000
simulation_ca_m_0125_distances<-sum(simulateddmaleindistances>median(ca_close_relatives_males_pairwise_distances))/10000
simulation_ca_difference_0125_distances<-sum(simulateddifferencesindistances>observeddifferenceindistances)/10000





```





```{r dispersal_analysisiii, eval=FALSE, warning=FALSE, results='asis', echo=TRUE}
options(width=60)

#Analysis 3: Correlogram to assess change of relatedness with distances


# Start with Arizona:

#have each value only once in the distance matrix
for (i in 1:ncol(AZ_all_pairwise_distance)) {
  AZ_all_pairwise_distance[i,i:ncol(AZ_all_pairwise_distance)]<-NA
}



#turn pairwise_r$wang into a matrix
az_all_relatedness<-select(az_pairwise_r,ind1.id,ind2.id,wang)
az_relatedness_matrix<-spread(az_all_relatedness,"ind1.id","wang")
az_relatedness_matrix <- cbind(az_relatedness_matrix,AF_061PR="NA")
az_relatedness_matrix<-arrange(az_relatedness_matrix,ind2.id)
az_relatedness_matrix <- InsertRow(data=az_relatedness_matrix, NewRow=rep("NA",samplesize[3,3]+samplesize[4,3]), RowNum=1)
az_relatedness_matrix[1,1] <- "AF_001YP"
rownames(az_relatedness_matrix)<-az_relatedness_matrix[,1]

#turn pairwise_r$quellergt into a matrix
az_relatedness_matrix<-select(az_pairwise_r,ind1.id,ind2.id,quellergt)
az_relatedness_matrix<-spread(az_relatedness_matrix,"ind1.id","quellergt")
az_relatedness_matrix <- cbind(az_relatedness_matrix,AF_061PR="NA")
az_relatedness_matrix<-arrange(az_relatedness_matrix,ind2.id)
az_relatedness_matrix <- InsertRow(data=az_relatedness_matrix, NewRow=rep("NA",samplesize[3,3]+samplesize[4,3]), RowNum=1)
az_relatedness_matrix[1,1] <- "AF_001YP"
rownames(az_relatedness_matrix)<-az_relatedness_matrix[,1]

relatedness_matrix<-relatedness_matrix[1:52,2:53]

female_relatedness_matrix<-relatedness_matrix[rownames(relatedness_matrix) %in% females,colnames(relatedness_matrix) %in% females]
male_relatedness_matrix<-relatedness_matrix[rownames(relatedness_matrix) %in% males,colnames(relatedness_matrix) %in% males]

#perform the correlogram analysis
#first way, defining the distance classes
female_correlogram_setdistances<-mantel.correlog(D.eco=female_relatedness_matrix,D.geo=female_pairwise_distances,break.pts=c(0,100,200,300,400,500,750,1250,1550,2000,2500),cutoff=FALSE,nperm=10000)
male_correlogram_setdistances<-mantel.correlog(D.eco=male_relatedness_matrix,D.geo=male_pairwise_distances,break.pts=c(0,100,200,300,400,500,750,1250,1550,2000,2500),cutoff=FALSE,nperm=10000)
#second way, setting the number of distance classes
female_correlogram_classes<-mantel.correlog(D.eco=female_relatedness_matrix,D.geo=female_pairwise_distances,n.class=5)
male_correlogram_classes<-mantel.correlog(D.eco=male_relatedness_matrix,D.geo=male_pairwise_distances,n.class=5)

#additional way, with the distance classes based on the inferred distance among relatives from analysis ii
female_correlogram_setdistances<-mantel.correlog(D.eco=female_relatedness_matrix,D.geo=female_pairwise_distances,break.pts=c(0,150,450,900,1600,2000),cutoff=FALSE,nperm=10000)
male_correlogram_setdistances<-mantel.correlog(D.eco=male_relatedness_matrix,D.geo=male_pairwise_distances,break.pts=c(0,150,450,900,1600,2000),cutoff=FALSE,nperm=10000)

female_correlogram_setdistances
male_correlogram_setdistances

```


### F. ETHICS

This research is carried out in accordance with permits from the:

1)  US Fish and Wildlife Service (scientific collecting permit number MB76700A-0,1,2)
2)  US Geological Survey Bird Banding Laboratory (federal bird banding permit number 23872)
3)  Arizona Game and Fish Department (scientific collecting license number SP594338 \[2017\], SP606267 \[2018\], SP639866 \[2019\], and SP402153 \[2020\])
4)  Institutional Animal Care and Use Committee at Arizona State University (protocol number 17-1594R)
5)  California Department of Fish and Wildlife (scientific collecting permit \[specific use\] number S‐192100001‐19210‐001)

### G. AUTHOR CONTRIBUTIONS

**Lukas:** Hypothesis development, data analysis and interpretation, write up, revising/editing.

**Logan:** Hypothesis development, data collection, data analysis and interpretation, write up, revising/editing, materials/funding.

### H. FUNDING

This research is funded by the Department of Human Behavior, Ecology and Culture at the Max Planck Institute for Evolutionary Anthropology.

### I. CONFLICT OF INTEREST DISCLOSURE

We, the authors, declare that we have no financial conflicts of interest with the content of this article. CJ Logan and D Lukas are Recommenders at PCI Ecology, and CJ Logan is on the Managing Board at PCI Ecology.

### J. ACKNOWLEDGEMENTS

We thank Kristine Johnson for technical advice on great-tailed grackles; Julia Cissewski and Sophie Kaube for tirelessly solving problems involving financial transactions and contracts; Richard McElreath for project support; and Aaron Blackwell and Ken Kosik for being the UCSB sponsors of the Cooperation Agreement with the Max Planck Institute for Evolutionary Anthropology.

### K. REFERENCES
